--- a/ktorrent/ktorrent.desktop.orig	2014-01-21 15:25:06.019687665 -0500
+++ b/ktorrent/ktorrent.desktop	2014-01-21 15:59:02.814273653 -0500
@@ -117,7 +117,7 @@
 Icon=ktorrent
 Type=Application
 X-DocPath=ktorrent/index.html
-MimeType=application/x-bittorrent;application/x-torrent;
+MimeType=application/x-bittorrent;application/x-torrent;x-scheme-handler/magnet;
 X-DBUS-StartupType=Unique
 X-DBUS-ServiceName=org.ktorrent.ktorrent
 Comment=A BitTorrent program for KDE
