diff -Naur apper-0.7.1-orig//ApperKCM/Settings/Settings.cpp apper-0.7.1/ApperKCM/Settings/Settings.cpp
--- apper-0.7.1-orig//ApperKCM/Settings/Settings.cpp	2012-02-21 14:22:05.000000000 +0100
+++ apper-0.7.1/ApperKCM/Settings/Settings.cpp	2012-03-01 18:34:52.245742576 +0100
@@ -61,6 +61,7 @@
         originTV->setEnabled(false);
         showOriginsCB->setEnabled(false);
     }
+    showOriginsCB->hide();
 
     intervalCB->addItem(i18nc("Hourly refresh the package cache", "Hourly"),  Enum::Hourly);
     intervalCB->addItem(i18nc("Daily refresh the package cache", "Daily"),   Enum::Daily);
