--- a/src/libopencv.h.cmake.in	2015-09-06 17:15:49.000000000 -0400
+++ b/src/libopencv.h.cmake.in	2015-09-06 17:47:06.000000000 -0400
@@ -51,14 +51,8 @@
 #define OPENCV_VERSION                         OPENCV_MAKE_VERSION(CV_MAJOR_VERSION,CV_MINOR_VERSION,CV_SUBMINOR_VERSION)
 #define OPENCV_TEST_VERSION(major,minor,patch) ( OPENCV_VERSION < OPENCV_MAKE_VERSION(major,minor,patch) )
 
-#if OPENCV_TEST_VERSION(3,0,0)
 #   include <opencv2/core/core.hpp>
-#   include <opencv2/core/internal.hpp>
-#   include <opencv2/contrib/contrib.hpp>
-#else
-#   include <opencv2/face.hpp>
 #   include <opencv2/core.hpp>
-#endif
 
 // for old-style code
 #if OPENCV_VERSION <= OPENCV_MAKE_VERSION(2,4,99)
