--- a/shell/jsoptparse.cpp	2013-02-11 23:33:23.000000000 +0100
+++ b/shell/jsoptparse.cpp	2019-02-03 22:43:24.956180457 +0100
@@ -243,7 +243,7 @@
     char *eq = strchr(argv[*i], '=');
     if (eq) {
         *value = eq + 1;
-        if (value[0] == '\0')
+        if (value[0][0] == '\0')
             return error("A value is required for option %.*s", eq - argv[*i], argv[*i]);
         return Okay;
     }
