--- CMakeLists.txt	(revision 1299643)
+++ CMakeLists.txt	(working copy)
@@ -29,6 +29,9 @@
    ENDIF(EXISTS ${_fullPath}/CMakeLists.txt)
 ENDMACRO (KDESDK_OPTIONAL_ADD_SUBDIRECTORY)
 
+cmake_minimum_required(VERSION 2.6)
+project(kdesdk)
+
 # list of self-contained submodules
 kdesdk_optional_add_subdirectory(kapptemplate)
 kdesdk_optional_add_subdirectory(dolphin-plugins)
@@ -43,9 +46,6 @@
 
 ############################
 # START OF OLD MODULE SETUP
-cmake_minimum_required(VERSION 2.6)
-project(kdesdk)
-
 set(CMAKE_MODULE_PATH ${CMAKE_SOURCE_DIR}/cmake/modules )
