commit 7f9693746f437da98c8b3a5d229ea8a1f0185842
Author: Marco Martin <notmart@gmail.com>
Date:   Thu Oct 16 10:45:30 2014 +0200

    crash--
    
    BUG:339875
    Change-Id: Ic8ab63853fa9b58dd83e1aa868e9599b0e37f09e

diff --git a/src/plasmapkg/plasmapkg.cpp b/src/plasmapkg/plasmapkg.cpp
index 7d55d0f..aed21e9 100644
--- a/src/plasmapkg/plasmapkg.cpp
+++ b/src/plasmapkg/plasmapkg.cpp
@@ -146,7 +146,7 @@ void PlasmaPkg::runMain()
         } else {
             package.setPath(d->packageFile);
         }
-        if (package.isValid()) {
+        if (package.isValid() && package.metadata().isValid()) {
             serviceType = package.metadata().property("X-Plasma-ServiceType").toString();
         }
 
