diff --git a/src/scriptengines/qml/plasmoid/containmentinterface.h b/src/scriptengines/qml/plasmoid/containmentinterface.h
index c058f8358b4aa123749959a3de5b0667d7a1fecc..5fe96b8d67bca179eab9c6365109b4bcae7f71cf 100644
--- a/src/scriptengines/qml/plasmoid/containmentinterface.h
+++ b/src/scriptengines/qml/plasmoid/containmentinterface.h
@@ -204,6 +204,7 @@ protected Q_SLOTS:
     void mimeTypeRetrieved(KIO::Job *job, const QString &mimetype);
 
 private Q_SLOTS:
+    Plasma::Applet *createApplet(const QString &plugin, const QVariantList &args, const QPoint &pos);
     Plasma::Applet *createApplet(const QString &plugin, const QVariantList &args, const QRectF &geom);
 
 private:
diff --git a/src/scriptengines/qml/plasmoid/containmentinterface.cpp b/src/scriptengines/qml/plasmoid/containmentinterface.cpp
index 4efc1e109bbab8ef43b686be25574ed5bd66d9ce..e7a43c721ada4d68086c104c54f5f50f5c0279b8 100644
--- a/src/scriptengines/qml/plasmoid/containmentinterface.cpp
+++ b/src/scriptengines/qml/plasmoid/containmentinterface.cpp
@@ -224,6 +224,11 @@ QRect ContainmentInterface::availableScreenRect() const
     return rect;
 }
 
+Plasma::Applet *ContainmentInterface::createApplet(const QString &plugin, const QVariantList &args, const QPoint &pos)
+{
+    return createApplet(plugin, args, QRectF(pos, QSize()));
+}
+
 Plasma::Applet *ContainmentInterface::createApplet(const QString &plugin, const QVariantList &args, const QRectF &geom)
 {
     //HACK
