--- a/src/VBox/Frontends/VirtualBox/src/globals/UIImageTools.cpp    2020-05-14 14:40:35.000000000 -0400
+++ b/src/VBox/Frontends/VirtualBox/src/globals/UIImageTools.cpp    2020-06-03 10:46:06.750776100 -0400
@@ -17,6 +17,7 @@
 
 /* Qt includes: */
 #include <QPainter>
+#include <QPainterPath>
 
 /* GUI include */
 #include "UIImageTools.h"
--- a/src/VBox/Frontends/VirtualBox/src/widgets/UIMenuToolBar.cpp   2020-05-14 14:40:44.000000000 -0400
+++ b/src/VBox/Frontends/VirtualBox/src/widgets/UIMenuToolBar.cpp   2020-06-03 10:52:50.125585673 -0400
@@ -19,6 +19,7 @@
 #include <QApplication>
 #include <QHBoxLayout>
 #include <QPainter>
+#include <QPainterPath>
 #include <QStyle>
 #include <QToolButton>
 
--- a/src/VBox/Frontends/VirtualBox/src/widgets/UIMiniToolBar.cpp   2020-05-14 14:40:44.000000000 -0400
+++ b/src/VBox/Frontends/VirtualBox/src/widgets/UIMiniToolBar.cpp   2020-06-03 10:51:38.560369848 -0400
@@ -21,6 +21,7 @@
 #include <QMenu>
 #include <QMoveEvent>
 #include <QPainter>
+#include <QPainterPath>
 #include <QStateMachine>
 #include <QStyle>
 #include <QTimer>
--- a/src/VBox/Frontends/VirtualBox/src/widgets/UIPopupBox.cpp      2020-05-14 14:40:44.000000000 -0400
+++ b/src/VBox/Frontends/VirtualBox/src/widgets/UIPopupBox.cpp      2020-06-03 10:47:51.348523624 -0400
@@ -19,6 +19,7 @@
 #include <QApplication>
 #include <QLabel>
 #include <QPainter>
+#include <QPainterPath>
 #include <QPaintEvent>
 #include <QStyle>
 #include <QVBoxLayout>
--- a/src/VBox/Frontends/VirtualBox/src/widgets/UIPopupBox.h        2020-05-14 14:40:44.000000000 -0400
+++ b/src/VBox/Frontends/VirtualBox/src/widgets/UIPopupBox.h        2020-06-03 10:49:01.201695063 -0400
@@ -24,6 +24,7 @@
 /* Qt includes: */
 #include <QIcon>
 #include <QWidget>
+#include <QPainterPath>
 
 /* GUI includes: */
 #include "UILibraryDefs.h"
