--- a/layout-templates/org.kde.plasma.desktop.defaultPanel/contents/layout.js	2021-07-29 16:46:20.507709968 +0200
+++ b/layout-templates/org.kde.plasma.desktop.defaultPanel/contents/layout.js	2021-07-29 16:46:20.577710663 +0200
@@ -27,7 +27,7 @@
 // For an Icons-Only Task Manager on the bottom, *3 is too much, *2 is too little
 // Round down to next highest even number since the Panel size widget only displays
 // even numbers
-panel.height = 2 * Math.floor(gridUnit * 2.5 / 2)
+panel.height = 44 // 2 * Math.floor(gridUnit * 2.5 / 2)
 
 // Restrict horizontal panel to a maximum size of a 21:9 monitor
 const maximumAspectRatio = 21/9;
