--- a/src/qtractorCurve.h	2015-07-24 18:06:44.000000000 -0400
+++ b/src/qtractorCurve.h	2015-07-28 16:03:23.517307956 -0400
@@ -26,6 +26,7 @@
 #include "qtractorMidiSequence.h"
 
 #include <QColor>
+#include <QObject>
 
 
 // Forward declarations.
--- a/src/qtractorMidiEngine.h	2015-07-24 18:06:44.000000000 -0400
+++ b/src/qtractorMidiEngine.h	2015-07-28 16:12:38.531415646 -0400
@@ -30,6 +30,7 @@
 #include <alsa/asoundlib.h>
 
 #include <QHash>
+#include <QObject>
 
 // Forward declarations.
 class qtractorMidiBus;
--- a/src/qtractorMidiTimer.cpp	2015-07-24 18:06:44.000000000 -0400
+++ b/src/qtractorMidiTimer.cpp	2015-07-28 16:17:12.551867711 -0400
@@ -24,6 +24,7 @@
 #include <alsa/asoundlib.h>
 
 #include <stdio.h>
+#include <QObject>
 
 
 //----------------------------------------------------------------------
--- a/src/qtractorTimeScale.cpp	2015-07-24 18:06:44.000000000 -0400
+++ a/src/qtractorTimeScale.cpp	2015-07-28 16:27:36.731125204 -0400
@@ -20,6 +20,7 @@
 *****************************************************************************/
 
 #include "qtractorTimeScale.h"
+#include <QObject>
 
 
 //----------------------------------------------------------------------
