--- a/src/ractionthreadbase_p.h	2014-08-21 18:35:33.975554315 -0400
+++ b/src/ractionthreadbase_p.h	2014-08-21 18:35:34.285563251 -0400
@@ -36,10 +36,9 @@
 
 // KDE includes
 
-#include <threadweaver/Job.h>
-#include <threadweaver/WeaverObserver.h>
-#include <threadweaver/State.h>
-#include <threadweaver/Thread.h>
+#include <threadweaver/job.h>
+#include <threadweaver/state.h>
+#include <threadweaver/thread.h>
 
 // Local includes
 
