--- a/src/ractionthreadbase.cpp	2014-08-21 18:36:17.506808877 -0400
+++ b/src/ractionthreadbase.cpp	2014-08-21 18:36:17.826818099 -0400
@@ -37,9 +37,9 @@
 #include <kdebug.h>
 #include <ThreadWeaver/JobCollection>
 #include <ThreadWeaver/Weaver>
-#include <threadweaver/ThreadWeaver.h>
-#include <threadweaver/Job.h>
-#include <threadweaver/DebuggingAids.h>
+#include <threadweaver/threadweaver.h>
+#include <threadweaver/job.h>
+#include <threadweaver/debuggingaids.h>
 #include <solid/device.h>
 
 // Local includes
