diff --git a/src/libtomahawk/playlist/TrackView.cpp b/src/libtomahawk/playlist/TrackView.cpp
index fd8241d..4c3b704 100644
--- a/src/libtomahawk/playlist/TrackView.cpp
+++ b/src/libtomahawk/playlist/TrackView.cpp
@@ -192,9 +192,7 @@ TrackView::setProxyModel( PlayableProxyModel* model )
     connect( m_proxyModel, SIGNAL( selectRequest( QPersistentModelIndex ) ), SLOT( select( QPersistentModelIndex ) ) );
     connect( m_proxyModel, SIGNAL( currentIndexChanged( QModelIndex, QModelIndex ) ), SLOT( onCurrentIndexChanged( QModelIndex, QModelIndex ) ) );
 
-    m_delegate = new PlaylistItemDelegate( this, m_proxyModel );
-    QTreeView::setItemDelegate( m_delegate );
-    QTreeView::setModel( m_proxyModel );
+    setPlaylistItemDelegate( new PlaylistItemDelegate( this, m_proxyModel ) );
 }
 
 
@@ -384,6 +382,10 @@ TrackView::currentChanged( const QModelIndex& current, const QModelIndex& previo
 {
     QTreeView::currentChanged( current, previous );
 
+    if(!m_model || !m_proxyModel) {
+        return;
+    }
+
     PlayableItem* item = m_model->itemFromIndex( m_proxyModel->mapToSource( current ) );
     if ( item && item->query() )
     {
