--- a/lib/KActivitiesConfig.cmake.in
+++ b/lib/KActivitiesConfig.cmake.in
@@ -1,6 +1,8 @@
 get_filename_component(myDir ${CMAKE_CURRENT_LIST_FILE} PATH) # get the directory where I myself am
 get_filename_component(rootDir ${myDir}/@relInstallDir@ ABSOLUTE) # get the chosen install prefix
 
+set(KACTIVITIES_FOUND TRUE)
+
 # set the version of myself
 set(KACTIVITIES_VERSION_MAJOR @KACTIVITIES_LIB_MAJOR_VERSION@)
 set(KACTIVITIES_VERSION_MINOR @KACTIVITIES_LIB_MINOR_VERSION@)
