--- a/src/main.cpp	2023-03-31 13:24:28.761219649 +0200
+++ b/src/mania.cpp	2023-03-31 13:24:28.829220443 +0200
@@ -52,15 +52,15 @@
 {
 #ifndef Q_OS_WIN
     // Prohibit using sudo or kdesu (but allow using the root user directly)
-    if (getuid() == 0) {
-        if (!qEnvironmentVariableIsEmpty("SUDO_USER")) {
-            std::cout << "Running Dolphin with sudo can cause bugs and expose you to security vulnerabilities." << std::endl;
-            return EXIT_FAILURE;
-        } else if (!qEnvironmentVariableIsEmpty("KDESU_USER")) {
-            std::cout << "Running Dolphin with kdesu can cause bugs and expose you to security vulnerabilities." << std::endl;
-            return EXIT_FAILURE;
-        }
-    }
+    //if (getuid() == 0) {
+    //    if (!qEnvironmentVariableIsEmpty("SUDO_USER")) {
+    //        std::cout << "Running Dolphin with sudo can cause bugs and expose you to security vulnerabilities." << std::endl;
+    //        return EXIT_FAILURE;
+    //    } else if (!qEnvironmentVariableIsEmpty("KDESU_USER")) {
+    //        std::cout << "Running Dolphin with kdesu can cause bugs and expose you to security vulnerabilities." << std::endl;
+    //        return EXIT_FAILURE;
+    //    }
+    //}
 #endif
 
     /**
