diff -Naur MediaInfo/Project/QMake/GUI/MediaInfoQt.pro MediaInfo-p/Project/QMake/GUI/MediaInfoQt.pro
--- MediaInfo/Project/QMake/GUI/MediaInfoQt.pro	2016-06-30 16:43:32.000000000 +0200
+++ MediaInfo-p/Project/QMake/GUI/MediaInfoQt.pro	2016-07-15 18:07:49.972662623 +0200
@@ -1,17 +1,19 @@
 # -------------------------------------------------
 # Project created by QtCreator 2010-07-23T13:03:11
 # -------------------------------------------------
-TARGET = mediainfo-gui
+TARGET = mediainfo-qt5-gui
 TEMPLATE = app
 
-# QT += webkit
-QT += xml
+QT += core xml widgets gui
+
+CONFIG += link_pkgconfig
+PKGCONFIG += libmediainfo libzen
+
 DEFINES += \
     _UNICODE
-LIBS += -ldl \
-    ../../../../MediaInfoLib/Project/GNU/Library/.libs/libmediainfo.a \
-    ../../../../ZenLib/Project/GNU/Library/.libs/libzen.a \
-    -lz
+
+LIBS += -ldl 
+   
 SOURCES += ../../../Source/GUI/Qt/main.cpp \
     ../../../Source/GUI/Qt/mainwindow.cpp \
     ../../../Source/Common/Core.cpp \
diff -Naur MediaInfo/Source/GUI/Qt/about.cpp MediaInfo-p/Source/GUI/Qt/about.cpp
--- MediaInfo/Source/GUI/Qt/about.cpp	2016-06-30 16:43:32.000000000 +0200
+++ MediaInfo-p/Source/GUI/Qt/about.cpp	2016-07-15 18:02:39.228658017 +0200
@@ -6,7 +6,7 @@
 
 #include "translate.h"
 #include "about.h"
-#include "_Automated/ui_about.h"
+#include "ui_about.h"
 #include <QtGui/QDesktopServices>
 #include <QtCore/QUrl>
 
diff -Naur MediaInfo/Source/GUI/Qt/about.h MediaInfo-p/Source/GUI/Qt/about.h
--- MediaInfo/Source/GUI/Qt/about.h	2016-06-30 16:43:32.000000000 +0200
+++ MediaInfo-p/Source/GUI/Qt/about.h	2016-07-15 18:02:39.228658017 +0200
@@ -7,7 +7,7 @@
 #ifndef ABOUT_H
 #define ABOUT_H
 
-#include <QtGui/QDialog>
+#include <QtWidgets/QDialog>
 #include <QtCore/QDebug>
 
 namespace Ui {
diff -Naur MediaInfo/Source/GUI/Qt/columneditsheet.cpp MediaInfo-p/Source/GUI/Qt/columneditsheet.cpp
--- MediaInfo/Source/GUI/Qt/columneditsheet.cpp	2016-06-30 16:43:32.000000000 +0200
+++ MediaInfo-p/Source/GUI/Qt/columneditsheet.cpp	2016-07-15 18:02:39.228658017 +0200
@@ -6,10 +6,10 @@
 
 #include "columneditsheet.h"
 #include "sheet.h"
-#include <QtGui/QLineEdit>
-#include <QtGui/QSpinBox>
-#include <QtGui/QComboBox>
-#include <QtGui/QToolButton>
+#include <QtWidgets/QLineEdit>
+#include <QtWidgets/QSpinBox>
+#include <QtWidgets/QComboBox>
+#include <QtWidgets/QToolButton>
 #include "translate.h"
 
 #include <ZenLib/Ztring.h>
@@ -23,7 +23,7 @@
     QHBoxLayout(parent),
     pos(pos),
     C(C),
-    col(c),
+    col(c)
 {
     QLineEdit* lineedit = new QLineEdit(c.name);
     this->addWidget(lineedit);
diff -Naur MediaInfo/Source/GUI/Qt/columneditsheet.h MediaInfo-p/Source/GUI/Qt/columneditsheet.h
--- MediaInfo/Source/GUI/Qt/columneditsheet.h	2016-06-30 16:43:32.000000000 +0200
+++ MediaInfo-p/Source/GUI/Qt/columneditsheet.h	2016-07-15 18:02:39.229658017 +0200
@@ -8,10 +8,10 @@
 #define COLUMNEDITSHEET_H
 
 #include "sheet.h"
-#include <QtGui/QHBoxLayout>
-#include <QtGui/QToolButton>
-#include <QtGui/QComboBox>
-#include <QtGui/QSpinBox>
+#include <QtWidgets/QHBoxLayout>
+#include <QtWidgets/QToolButton>
+#include <QtWidgets/QComboBox>
+#include <QtWidgets/QSpinBox>
 #include <QtCore/QDebug>
 
 class ColumnEditSheet : public QHBoxLayout
diff -Naur MediaInfo/Source/GUI/Qt/configtreetext.h MediaInfo-p/Source/GUI/Qt/configtreetext.h
--- MediaInfo/Source/GUI/Qt/configtreetext.h	2016-06-30 16:43:32.000000000 +0200
+++ MediaInfo-p/Source/GUI/Qt/configtreetext.h	2016-07-15 18:02:39.229658017 +0200
@@ -11,7 +11,7 @@
 #include <QtCore/QString>
 #include <QtCore/QStringList>
 #include <QtCore/QSettings>
-#include <QtGui/QComboBox>
+#include <QtWidgets/QComboBox>
 #include <QtCore/QDebug>
 
 class ConfigTreeText
diff -Naur MediaInfo/Source/GUI/Qt/custom.h MediaInfo-p/Source/GUI/Qt/custom.h
--- MediaInfo/Source/GUI/Qt/custom.h	2016-06-30 16:43:32.000000000 +0200
+++ MediaInfo-p/Source/GUI/Qt/custom.h	2016-07-15 18:02:39.229658017 +0200
@@ -11,7 +11,7 @@
 #include <QtCore/QString>
 #include <QtCore/QStringList>
 #include <QtCore/QSettings>
-#include <QtGui/QComboBox>
+#include <QtWidgets/QComboBox>
 #include "Common/Core.h"
 #include <QtCore/QDebug>
 
diff -Naur MediaInfo/Source/GUI/Qt/easyviewwidget.cpp MediaInfo-p/Source/GUI/Qt/easyviewwidget.cpp
--- MediaInfo/Source/GUI/Qt/easyviewwidget.cpp	2016-06-30 16:43:32.000000000 +0200
+++ MediaInfo-p/Source/GUI/Qt/easyviewwidget.cpp	2016-07-15 18:02:39.230658017 +0200
@@ -5,10 +5,10 @@
  */
 
 #include "easyviewwidget.h"
-#include <QtGui/QGroupBox>
-#include <QtGui/QLabel>
-#include <QtGui/QBoxLayout>
-#include <QtGui/QComboBox>
+#include <QtWidgets/QGroupBox>
+#include <QtWidgets/QLabel>
+#include <QtWidgets/QBoxLayout>
+#include <QtWidgets/QComboBox>
 
 #include <ZenLib/Ztring.h>
 using namespace ZenLib;
diff -Naur MediaInfo/Source/GUI/Qt/easyviewwidget.h MediaInfo-p/Source/GUI/Qt/easyviewwidget.h
--- MediaInfo/Source/GUI/Qt/easyviewwidget.h	2016-06-30 16:43:32.000000000 +0200
+++ MediaInfo-p/Source/GUI/Qt/easyviewwidget.h	2016-07-15 18:02:39.230658017 +0200
@@ -7,9 +7,9 @@
 #ifndef EASYVIEWWIDGET_H
 #define EASYVIEWWIDGET_H
 
-#include <QtGui/QWidget>
+#include <QtWidgets/QWidget>
 #include "Common/Core.h"
-#include <QtGui/QGroupBox>
+#include <QtWidgets/QGroupBox>
 #include <QtCore/QDebug>
 
 class EasyViewWidget : public QWidget
diff -Naur MediaInfo/Source/GUI/Qt/editconfigtreetext.cpp MediaInfo-p/Source/GUI/Qt/editconfigtreetext.cpp
--- MediaInfo/Source/GUI/Qt/editconfigtreetext.cpp	2016-06-30 16:43:32.000000000 +0200
+++ MediaInfo-p/Source/GUI/Qt/editconfigtreetext.cpp	2016-07-15 18:02:39.230658017 +0200
@@ -5,11 +5,11 @@
  */
 
 #include "editconfigtreetext.h"
-#include "_Automated/ui_editconfigtreetext.h"
+#include "ui_editconfigtreetext.h"
 
 #include "configtreetext.h"
 #include "Common/Core.h"
-#include <QtGui/QCheckBox>
+#include <QtWidgets/QCheckBox>
 
 #include <ZenLib/Ztring.h>
 using namespace ZenLib;
diff -Naur MediaInfo/Source/GUI/Qt/editconfigtreetext.h MediaInfo-p/Source/GUI/Qt/editconfigtreetext.h
--- MediaInfo/Source/GUI/Qt/editconfigtreetext.h	2016-06-30 16:43:32.000000000 +0200
+++ MediaInfo-p/Source/GUI/Qt/editconfigtreetext.h	2016-07-15 18:02:39.231658017 +0200
@@ -8,7 +8,7 @@
 #define EDITCONFIGTREETEXT_H
 
 #include "translate.h"
-#include <QtGui/QDialog>
+#include <QtWidgets/QDialog>
 #include "Common/Core.h"
 #include "configtreetext.h"
 #include <QtCore/QDebug>
diff -Naur MediaInfo/Source/GUI/Qt/editcustom.cpp MediaInfo-p/Source/GUI/Qt/editcustom.cpp
--- MediaInfo/Source/GUI/Qt/editcustom.cpp	2016-06-30 16:43:32.000000000 +0200
+++ MediaInfo-p/Source/GUI/Qt/editcustom.cpp	2016-07-15 18:02:39.231658017 +0200
@@ -6,9 +6,9 @@
 
 #include "translate.h"
 #include "editcustom.h"
-#include "_Automated/ui_editcustom.h"
+#include "ui_editcustom.h"
 
-#include <QtGui/QTextEdit>
+#include <QtWidgets/QTextEdit>
 
 #include <ZenLib/Ztring.h>
 using namespace ZenLib;
diff -Naur MediaInfo/Source/GUI/Qt/editcustom.h MediaInfo-p/Source/GUI/Qt/editcustom.h
--- MediaInfo/Source/GUI/Qt/editcustom.h	2016-06-30 16:43:32.000000000 +0200
+++ MediaInfo-p/Source/GUI/Qt/editcustom.h	2016-07-15 18:02:39.231658017 +0200
@@ -7,7 +7,7 @@
 #ifndef EDITCUSTOM_H
 #define EDITCUSTOM_H
 
-#include <QtGui/QDialog>
+#include <QtWidgets/QDialog>
 #include "Common/Core.h"
 #include "custom.h"
 
diff -Naur MediaInfo/Source/GUI/Qt/editsheet.cpp MediaInfo-p/Source/GUI/Qt/editsheet.cpp
--- MediaInfo/Source/GUI/Qt/editsheet.cpp	2016-06-30 16:43:32.000000000 +0200
+++ MediaInfo-p/Source/GUI/Qt/editsheet.cpp	2016-07-15 18:02:39.231658017 +0200
@@ -6,17 +6,17 @@
 
 #include "translate.h"
 #include "editsheet.h"
-#include "_Automated/ui_editsheet.h"
+#include "ui_editsheet.h"
 #include "sheet.h"
 #include "columneditsheet.h"
-#include "QtGui/QHBoxLayout"
-#include "QtGui/QLabel"
-#include "QtGui/QLineEdit"
-#include "QtGui/QSpinBox"
-#include "QtGui/QPushButton"
-#include "QtGui/QToolButton"
-#include "QtGui/QSizePolicy"
-#include "QtGui/QComboBox"
+#include "QtWidgets/QHBoxLayout"
+#include "QtWidgets/QLabel"
+#include "QtWidgets/QLineEdit"
+#include "QtWidgets/QSpinBox"
+#include "QtWidgets/QPushButton"
+#include "QtWidgets/QToolButton"
+#include "QtWidgets/QSizePolicy"
+#include "QtWidgets/QComboBox"
 
 EditSheet::EditSheet(Sheet* s, Core* C, QWidget *parent) :
     QDialog(parent),
diff -Naur MediaInfo/Source/GUI/Qt/editsheet.h MediaInfo-p/Source/GUI/Qt/editsheet.h
--- MediaInfo/Source/GUI/Qt/editsheet.h	2016-06-30 16:43:32.000000000 +0200
+++ MediaInfo-p/Source/GUI/Qt/editsheet.h	2016-07-15 18:02:39.232658017 +0200
@@ -7,7 +7,7 @@
 #ifndef EDITSHEET_H
 #define EDITSHEET_H
 
-#include <QtGui/QDialog>
+#include <QtWidgets/QDialog>
 #include "sheet.h"
 #include <QtCore/QDebug>
 
diff -Naur MediaInfo/Source/GUI/Qt/export.cpp MediaInfo-p/Source/GUI/Qt/export.cpp
--- MediaInfo/Source/GUI/Qt/export.cpp	2016-06-30 16:43:32.000000000 +0200
+++ MediaInfo-p/Source/GUI/Qt/export.cpp	2016-07-15 18:02:39.232658017 +0200
@@ -6,9 +6,9 @@
 
 #include "export.h"
 #include "translate.h"
-#include "_Automated/ui_export.h"
-#include <QtGui/QFileDialog>
-#include <QtGui/QMessageBox>
+#include "ui_export.h"
+#include <QtWidgets/QFileDialog>
+#include <QtWidgets/QMessageBox>
 #include "configtreetext.h"
 #include "mainwindow.h"
 
diff -Naur MediaInfo/Source/GUI/Qt/export.h MediaInfo-p/Source/GUI/Qt/export.h
--- MediaInfo/Source/GUI/Qt/export.h	2016-06-30 16:43:32.000000000 +0200
+++ MediaInfo-p/Source/GUI/Qt/export.h	2016-07-15 18:02:39.232658017 +0200
@@ -7,7 +7,7 @@
 #ifndef EXPORT_H
 #define EXPORT_H
 
-#include <QtGui/QDialog>
+#include <QtWidgets/QDialog>
 #include <QtCore/QIODevice>
 #include <QtCore/QDebug>
 
diff -Naur MediaInfo/Source/GUI/Qt/main.cpp MediaInfo-p/Source/GUI/Qt/main.cpp
--- MediaInfo/Source/GUI/Qt/main.cpp	2016-06-30 16:43:32.000000000 +0200
+++ MediaInfo-p/Source/GUI/Qt/main.cpp	2016-07-15 18:02:39.233658017 +0200
@@ -4,7 +4,7 @@
  *  be found in the License.html file in the root of the source tree.
  */
 
-#include <QtGui/QApplication>
+#include <QtWidgets/QApplication>
 #include <QtCore/QtPlugin>
 #include <iostream>
 #include "mainwindow.h"
diff -Naur MediaInfo/Source/GUI/Qt/mainwindow.cpp MediaInfo-p/Source/GUI/Qt/mainwindow.cpp
--- MediaInfo/Source/GUI/Qt/mainwindow.cpp	2016-06-30 16:43:32.000000000 +0200
+++ MediaInfo-p/Source/GUI/Qt/mainwindow.cpp	2016-07-15 18:08:31.590663240 +0200
@@ -6,21 +6,22 @@
 
 #include "mainwindow.h"
 #include "translate.h"
-#include "_Automated/ui_mainwindow.h"
+#include "ui_mainwindow.h"
 
 #include <iostream>
 #include <iomanip>
+#include <QMimeData>
 #include <QtGui/QDropEvent>
-#include <QtGui/QFileDialog>
-#include <QtGui/QMessageBox>
+#include <QtWidgets/QFileDialog>
+#include <QtWidgets/QMessageBox>
 #include <QtCore/QDir>
-#include <QtGui/QActionGroup>
-#include <QtGui/QTextBrowser>
-#include <QtGui/QLabel>
+#include <QtWidgets/QActionGroup>
+#include <QtWidgets/QTextBrowser>
+#include <QtWidgets/QLabel>
 #include <QtCore/QSettings>
 #include <QtXml/QDomDocument>
-#include <QtGui/QTreeWidget>
-#include <QtGui/QToolButton>
+#include <QtWidgets/QTreeWidget>
+#include <QtWidgets/QToolButton>
 #include "easyviewwidget.h"
 #include "preferences.h"
 #include "about.h"
@@ -69,7 +70,6 @@
         ui->menuView->addAction(action);
         menuItemGroup->addAction(action);
     }
-    connect(menuItemGroup,SIGNAL(selected(QAction*)),SLOT(actionView_toggled(QAction*)));
     menuItemGroup->setParent(ui->menuView);
 
     QToolButton* tb = new QToolButton(ui->toolBar);
diff -Naur MediaInfo/Source/GUI/Qt/mainwindow.h MediaInfo-p/Source/GUI/Qt/mainwindow.h
--- MediaInfo/Source/GUI/Qt/mainwindow.h	2016-06-30 16:43:32.000000000 +0200
+++ MediaInfo-p/Source/GUI/Qt/mainwindow.h	2016-07-15 18:02:39.234658017 +0200
@@ -7,13 +7,13 @@
 #ifndef MAINWINDOW_H
 #define MAINWINDOW_H
 
-#include <QtGui/QMainWindow>
+#include <QtWidgets/QMainWindow>
 #include "Common/Core.h"
 #include <QtCore/QSettings>
 #include <QtCore/QDir>
-#include <QtGui/QTreeWidget>
-#include <QtGui/QTextBrowser>
-#include <QtGui/QProgressDialog>
+#include <QtWidgets/QTreeWidget>
+#include <QtWidgets/QTextBrowser>
+#include <QtWidgets/QProgressDialog>
 #include <QtCore/QTimer>
 #include "views.h"
 #include <QtCore/QDebug>
diff -Naur MediaInfo/Source/GUI/Qt/preferences.cpp MediaInfo-p/Source/GUI/Qt/preferences.cpp
--- MediaInfo/Source/GUI/Qt/preferences.cpp	2016-06-30 16:43:32.000000000 +0200
+++ MediaInfo-p/Source/GUI/Qt/preferences.cpp	2016-07-15 18:02:39.234658017 +0200
@@ -6,8 +6,8 @@
 
 #include "preferences.h"
 #include "translate.h"
-#include "_Automated/ui_preferences.h"
-#include <QtGui/QLabel>
+#include "ui_preferences.h"
+#include <QtWidgets/QLabel>
 #include "views.h"
 #include "sheet.h"
 #include "configtreetext.h"
diff -Naur MediaInfo/Source/GUI/Qt/preferences.h MediaInfo-p/Source/GUI/Qt/preferences.h
--- MediaInfo/Source/GUI/Qt/preferences.h	2016-06-30 16:43:32.000000000 +0200
+++ MediaInfo-p/Source/GUI/Qt/preferences.h	2016-07-15 18:02:39.234658017 +0200
@@ -7,7 +7,7 @@
 #ifndef PREFERENCES_H
 #define PREFERENCES_H
 
-#include <QtGui/QDialog>
+#include <QtWidgets/QDialog>
 #include <QtCore/QSettings>
 #include "Common/Core.h"
 #include <QtCore/QDebug>
diff -Naur MediaInfo/Source/GUI/Qt/sheetview.cpp MediaInfo-p/Source/GUI/Qt/sheetview.cpp
--- MediaInfo/Source/GUI/Qt/sheetview.cpp	2016-06-30 16:43:32.000000000 +0200
+++ MediaInfo-p/Source/GUI/Qt/sheetview.cpp	2016-07-15 18:02:39.235658017 +0200
@@ -6,7 +6,7 @@
 
 #include "sheetview.h"
 #include "translate.h"
-#include "_Automated/ui_sheetview.h"
+#include "ui_sheetview.h"
 #include "sheet.h"
 #include "mainwindow.h"
 #include <QtGui/QDesktopServices>
diff -Naur MediaInfo/Source/GUI/Qt/sheetview.h MediaInfo-p/Source/GUI/Qt/sheetview.h
--- MediaInfo/Source/GUI/Qt/sheetview.h	2016-06-30 16:43:32.000000000 +0200
+++ MediaInfo-p/Source/GUI/Qt/sheetview.h	2016-07-15 18:02:39.235658017 +0200
@@ -7,7 +7,7 @@
 #ifndef SHEETVIEW_H
 #define SHEETVIEW_H
 
-#include <QtGui/QFrame>
+#include <QtWidgets/QFrame>
 #include "Common/Core.h"
 #include <QtCore/QDebug>
 
