diff -Naur kdebase-workspace-4.4.92.orig/plasma/desktop/applets/kickoff/applet/applet.cpp kdebase-workspace-4.4.92/plasma/desktop/applets/kickoff/applet/applet.cpp
--- kdebase-workspace-4.4.92.orig/plasma/desktop/applets/kickoff/applet/applet.cpp     2010-07-07 19:10:16.000000000 +0200
+++ kdebase-workspace-4.4.92/plasma/desktop/applets/kickoff/applet/applet.cpp  2010-07-08 03:42:54.993128535 +0200
@@ -45,6 +45,9 @@
 #include "ui/launcher.h"
 #include "core/recentapplications.h"
 
+// Branding
+#include <QFile>
+
 class LauncherApplet::Private
 {
 public:
@@ -187,7 +190,10 @@
 void LauncherApplet::configChanged()
 {
     KConfigGroup cg = config();
-    setPopupIcon(cg.readEntry("icon", "start-here-kde"));
+    if (QFile::exists("/usr/share/icons/oxygen/scalable/places/start-here-branding.svgz"))
+      setPopupIcon(cg.readEntry("icon", "start-here-branding"));
+    else
+      setPopupIcon(cg.readEntry("icon", "start-here-kde"));
     constraintsEvent(Plasma::ImmutableConstraint);
 }
 
diff -Naur kdebase-workspace-4.4.92.orig/plasma/desktop/applets/kickoff/simpleapplet/simpleapplet.cpp kdebase-workspace-4.4.92/plasma/desktop/applets/kickoff/simpleapplet/simpleapplet.cpp
--- kdebase-workspace-4.4.92.orig/plasma/desktop/applets/kickoff/simpleapplet/simpleapplet.cpp       2010-07-07 19:10:15.000000000 +0200
+++ kdebase-workspace-4.4.92/plasma/desktop/applets/kickoff/simpleapplet/simpleapplet.cpp    2010-07-08 03:44:38.979784323 +0200
@@ -36,6 +36,7 @@
 #include <QtGui/QSpacerItem>
 #include <QtGui/QListWidget>
 #include <QtGui/QListWidgetItem>
+#include <QFile>
 
 // KDE Libs
 #include <KActionCollection>
@@ -74,6 +75,9 @@
 #include "core/urlitemlauncher.h"
 #include "ui/contextmenufactory.h"
 
+// Branding
+#include <QFile>
+
 #ifndef KDE_USE_FINAL
 Q_DECLARE_METATYPE(QPersistentModelIndex)
 #endif
@@ -305,7 +309,10 @@
                 d->viewtypes << "RunCommand";
             }
             d->viewtypes << "Leave";
-            d->iconname = "start-here-kde";
+            if (QFile::exists("/usr/share/icons/oxygen/scalable/places/start-here-branding.svgz"))
+              d->iconname = "start-here-branding";
+            else
+              d->iconname = "start-here-kde";
         }
     }
     d->formattype = NameDescription;
