--- plasma/desktop/applets/kickoff/ui/brandingbutton.cpp	2013-02-02 23:28:25.395227907 +0100
+++ plasma/desktop/applets/kickoff/ui/brandingbutton.cpp.new	2013-02-02 23:40:52.263704398 +0100
@@ -25,6 +25,7 @@
 #include <KDebug>
 #include <KStandardDirs>
 #include <KRun>
+#include <KDE/KUrl>
 
 #include <Plasma/Svg>
 #include <Plasma/Theme>
@@ -68,7 +69,7 @@
 
 void BrandingButton::openHomepage()
 {
-    new KRun(Plasma::Theme::defaultTheme()->homepage(), topLevelWidget(), false, false);
+    new KRun(KUrl("http://www.chakra-project.org"), topLevelWidget(), false, false);
 }
 
 void BrandingButton::paintEvent(QPaintEvent *event)
