diff -Naur kdebase-workspace-4.3.80.orig/plasma/desktop/applets/kickoff/ui/brandingbutton.cpp kdebase-workspace-4.3.80/plasma/desktop/applets/kickoff/ui/brandingbutton.cpp
--- kdebase-workspace-4.3.80.orig/plasma/desktop/applets/kickoff/ui/brandingbutton.cpp	2009-09-16 11:40:52.000000000 +0000
+++ kdebase-workspace-4.3.80/plasma/desktop/applets/kickoff/ui/brandingbutton.cpp	2009-12-06 18:23:21.000000000 +0000
@@ -69,7 +69,7 @@
 void BrandingButton::openHomepage()
 {
     //FIXME: 4.3 .. add a brandingConfig to Theme
-    KUrl home("http://www.kde.org");
+    KUrl home("http://www.chakra-project.org");
     QString themePath = KStandardDirs::locate("data", "desktoptheme/" +
                                               Plasma::Theme::defaultTheme()->themeName() +
                                               "/metadata.desktop");
