diff -Naur kdebase-workspace-4.7.80.orig/plasma/desktop/applets/kickoff/core/favoritesmodel.cpp 
kdebase-workspace-4.7.80/plasma/desktop/applets/kickoff/core/favoritesmodel.cpp
--- kdebase-workspace-4.7.80.orig/plasma/desktop/applets/kickoff/core/favoritesmodel.cpp	2009-09-16 11:40:52.000000000 +0000
+++ kdebase-workspace-4.7.80/plasma/desktop/applets/kickoff/core/favoritesmodel.cpp	2009-12-06 18:31:29.000000000 +0000
@@ -105,7 +105,7 @@
     static QList<QString> defaultFavorites()
     {
         QList<QString> applications;
-        applications << "konqbrowser" << "KMail2" << "systemsettings" << "dolphin";
+        applications << "dolphin" << "konqbrowser" << "kate" << "kopete" << "KMail" << "systemsettings" << "pkgmanager" << "konsole" << "k3b";
 
         QList<QString> desktopFiles;
 
