diff --git a/kcontrol/dateandtime/dtime.cpp b/kcontrol/dateandtime/dtime.cpp
index 518afe5..e0147d9 100644
--- a/kcontrol/dateandtime/dtime.cpp
+++ b/kcontrol/dateandtime/dtime.cpp
@@ -238,7 +238,6 @@ void Dtime::save( QVariantMap& helperargs )
   helperargs["ntp"] = true;
   helperargs["ntpServers"] = list;
   helperargs["ntpEnabled"] = setDateTimeAuto->isChecked();
-  helperargs["ntpUtility"] = ntpUtility;
 
   if(setDateTimeAuto->isChecked() && !ntpUtility.isEmpty()){
     // NTP Time setting - done in helper
diff --git a/kcontrol/dateandtime/helper.h b/kcontrol/dateandtime/helper.h
index 1cbf103..fea71ba 100644
--- a/kcontrol/dateandtime/helper.h
+++ b/kcontrol/dateandtime/helper.h
@@ -42,8 +42,7 @@ class ClockHelper : public QObject
         ActionReply save(const QVariantMap &map);
 
     private:
-        int ntp(const QStringList& ntpServers, bool ntpEnabled,
-                const QString& ntpUtility);
+        int ntp(const QStringList& ntpServers, bool ntpEnabled);
         int date(const QString& newdate, const QString& olddate);
         int tz(const QString& selectedzone);
         int tzreset();
diff --git a/kcontrol/dateandtime/helper.cpp b/kcontrol/dateandtime/helper.cpp
index 9168db3..87f3b96 100644
--- a/kcontrol/dateandtime/helper.cpp
+++ b/kcontrol/dateandtime/helper.cpp
@@ -52,8 +52,7 @@
 // clears it. So we have to use a reasonable default.
 static const QString exePath = QLatin1String("/usr/sbin:/usr/bin:/sbin:/bin");
 
-int ClockHelper::ntp( const QStringList& ntpServers, bool ntpEnabled,
-                      const QString& ntpUtility )
+int ClockHelper::ntp( const QStringList& ntpServers, bool ntpEnabled )
 {
   int ret = 0;
 
@@ -69,24 +68,30 @@ int ClockHelper::ntp( const QStringList& ntpServers, bool ntpEnabled,
   config.writeEntry("servers", ntpServers );
   config.writeEntry("enabled", ntpEnabled );
 
-  if ( ntpEnabled && !ntpUtility.isEmpty() ) {
-    // NTP Time setting
-    QString timeServer = ntpServers.first();
-    if( timeServer.indexOf( QRegExp(".*\\(.*\\)$") ) != -1 ) {
-      timeServer.replace( QRegExp(".*\\("), "" );
-      timeServer.replace( QRegExp("\\).*"), "" );
-      // Would this be better?: s/^.*\(([^)]*)\).*$/\1/
-    }
+  if ( ntpEnabled ) {
+    QString ntpUtility;
+    foreach(const QString &possible_ntputility, QStringList() << "ntpdate" << "rdate" ) {
+        ntpUtility = KStandardDirs::findExe(possible_ntputility);
+        if (!ntpUtility.isEmpty()) {
+            // NTP Time setting
+            QString timeServer = ntpServers.first();
+            if( timeServer.indexOf( QRegExp(".*\\(.*\\)$") ) != -1 ) {
+            timeServer.replace( QRegExp(".*\\("), "" );
+            timeServer.replace( QRegExp("\\).*"), "" );
+            // Would this be better?: s/^.*\(([^)]*)\).*$/\1/
+            }
 
-    KProcess proc;
-    proc << ntpUtility << timeServer;
-    if ( proc.execute() != 0 ) {
-      ret |= NTPError;
+            KProcess proc;
+            proc << ntpUtility << timeServer;
+            if ( proc.execute() != 0 ) {
+            ret |= NTPError;
+            }
+        } else if( ntpEnabled ) {
+            ret |= NTPError;
+        }
+        break;
     }
-  } else if( ntpEnabled ) {
-    ret |= NTPError;
   }
-
   return ret;
 }
 
@@ -111,6 +116,13 @@ int ClockHelper::date( const QString& newdate, const QString& olddate )
 int ClockHelper::tz( const QString& selectedzone )
 {
     int ret = 0;
+
+    //only allow letters, numbers hyphen underscore plus and forward slash
+    //allowed pattern taken from time-util.c in systemd
+    if (!QRegExp("[a-zA-Z0-9-_+/]*").exactMatch(selectedzone)) {
+        return ret;
+    }
+
 #if defined(USE_SOLARIS)	// MARCO
 
         KTemporaryFile tf;
@@ -227,7 +239,7 @@ ActionReply ClockHelper::save(const QVariantMap &args)
   int ret = 0; // error code
 //  The order here is important
   if( _ntp )
-    ret |= ntp( args.value("ntpServers").toStringList(), args.value("ntpEnabled").toBool(), args.value("ntpUtility").toString() );
+    ret |= ntp( args.value("ntpServers").toStringList(), args.value("ntpEnabled").toBool());
   if( _date )
     ret |= date( args.value("newdate").toString(), args.value("olddate").toString() );
   if( _tz )
