--- a/pykde4-4.9.97/CMakeLists.txt	2013-01-04 15:47:40.000000000 -0500
+++ b/pykde4-4.9.97/CMakeLists.txt	2013-01-04 13:59:38.571700297 -0500
@@ -1,14 +1,8 @@
 project(pykde4)
 
-cmake_minimum_required(VERSION 2.8.6 FATAL_ERROR)
-
 # search packages used by KDE
 find_package(KDE4 4.9.4 REQUIRED)
 
-# Match what's used in the main macros
-cmake_policy(SET CMP0002 OLD)
-find_package(PythonLibrary)
-
 
 
 include(KDE4Defaults)
