diff --git a/plugins/multimedia/pulseaudio/qaudioinput_pulse.cpp b/plugins/multimedia/pulseaudio/qaudioinput_pulse.cpp
index f916595..136f4da 100644
--- a/plugins/multimedia/pulseaudio/qaudioinput_pulse.cpp
+++ b/plugins/multimedia/pulseaudio/qaudioinput_pulse.cpp
@@ -47,6 +47,8 @@
 #include "qpulseaudioengine.h"
 #include "qpulsehelpers.h"
 
+#include <unistd.h>
+
 QT_BEGIN_NAMESPACE
 
 const int PeriodTimeMs = 50;
diff --git a/plugins/multimedia/pulseaudio/qaudiooutput_pulse.cpp b/plugins/multimedia/pulseaudio/qaudiooutput_pulse.cpp
index 90c4470..a48ab8c 100644
--- a/plugins/multimedia/pulseaudio/qaudiooutput_pulse.cpp
+++ b/plugins/multimedia/pulseaudio/qaudiooutput_pulse.cpp
@@ -47,6 +47,8 @@
 #include "qpulseaudioengine.h"
 #include "qpulsehelpers.h"
 
+#include <unistd.h>
+
 QT_BEGIN_NAMESPACE
 
 const int PeriodTimeMs = 20;
diff --git a/plugins/multimedia/pulseaudio/qpulseaudioengine.cpp b/plugins/multimedia/pulseaudio/qpulseaudioengine.cpp
index 1abf93b..05014d3 100644
--- a/plugins/multimedia/pulseaudio/qpulseaudioengine.cpp
+++ b/plugins/multimedia/pulseaudio/qpulseaudioengine.cpp
@@ -47,6 +47,8 @@
 #include "qaudiooutput_pulse.h"
 #include "qpulsehelpers.h"
 
+#include <unistd.h>
+
 QT_BEGIN_NAMESPACE
 
 static void serverInfoCallback(pa_context *context, const pa_server_info *info, void *userdata)
