--- src/lib/app/browserwindow.cpp.orig	2014-03-04 18:35:40.000211594 -0500
+++ src/lib/app/browserwindow.cpp	2014-03-04 18:35:40.573560173 -0500
@@ -758,7 +758,7 @@
 
     //Url settings
     settings.beginGroup("Web-URL-Settings");
-    m_homepage = settings.value("homepage", "qupzilla:start").toUrl();
+    m_homepage = settings.value("homepage", "qupzilla:speeddial").toUrl();
     settings.endGroup();
 
     QWebSettings* websettings = mApp->webSettings();
@@ -774,7 +774,7 @@
     bool showWebSearchBar = settings.value("showWebSearchBar", true).toBool();
     bool showBookmarksToolbar = settings.value("showBookmarksToolbar", true).toBool();
     bool showNavigationToolbar = settings.value("showNavigationToolbar", true).toBool();
-    bool showMenuBar = settings.value("showMenubar", true).toBool();
+    bool showMenuBar = settings.value("showMenubar", false).toBool();
     bool makeTransparent = settings.value("useTransparentBackground", false).toBool();
     m_sideBarWidth = settings.value("SideBarWidth", 250).toInt();
     m_webViewWidth = settings.value("WebViewWidth", 2000).toInt();
