--- a/src/mainwindow.cpp	2016-01-14 15:05:45.605040133 -0500
+++ b/src/mainwindow.cpp	2016-01-14 15:17:52.975347304 -0500
@@ -1551,6 +1551,19 @@
 
     tempFile->write(out);
     tempFile->flush();
+    
+    tempFile->write("\n\n");
+    
+    tempFile->write("----------------------------------------------------------------------------------------------------------\n");
+    tempFile->write("cat /var/log/installation.log\n");
+    tempFile->write("----------------------------------------------------------------------------------------------------------\n\n");
+    out = UnixCommand::getCommandOutput("cat /var/log/installation.log");
+
+    //out.replace(hostname, "<HOSTNAME>");
+    out.replace(homePath, "<HOME_PATH>");
+
+    tempFile->write(out);
+    tempFile->flush();
     tempFile->close();
   }
   else
