--- src/main.cpp.orig	2014-01-26 17:55:48.000000000 -0500
+++ src/main.cpp	2014-01-26 20:24:36.293037490 -0500
@@ -29,7 +29,7 @@
 #include <QtGui>
 #include <QMessageBox>
 
-//#define NO_GTK_STYLE
+#define NO_GTK_STYLE
 
 int main(int argc, char *argv[])
 {

--- octopi.pro.orig	2014-01-26 17:55:48.000000000 -0500
+++ octopi.pro	2014-01-27 17:17:05.140903743 -0500
@@ -85,6 +85,7 @@
     resources/translations/octopi_fr_FR.ts \
     resources/translations/octopi_es_ES.ts \
     resources/translations/octopi_es_419.ts \
+    resources/translations/octopi_es_AR.ts \
     resources/translations/octopi_el.ts \
     resources/translations/octopi_de.ts \
     resources/translations/octopi_it.ts \

--- resources.qrc.orig	2014-01-26 17:55:48.000000000 -0500
+++ resources.qrc	2014-01-27 17:33:40.534273806 -0500
@@ -47,6 +47,7 @@
         <file>resources/translations/octopi_pt_BR.qm</file>
         <file>resources/translations/octopi_fr_FR.qm</file>
         <file>resources/translations/octopi_es_ES.qm</file>
+	<file>resources/translations/octopi_es_AR.qm</file>
         <file>resources/translations/octopi_el.qm</file>
         <file>resources/translations/octopi_de.qm</file>
         <file>resources/translations/octopi_it.qm</file>
