--- src/main.cpp.orig	2013-07-20 13:08:21.000000000 -0400
+++ src/main.cpp	2013-07-20 13:22:30.119234883 -0400
@@ -32,33 +32,6 @@
   ArgumentList *argList = new ArgumentList(argc, argv);
   QApplication::setGraphicsSystem(QLatin1String("raster"));
 
-  if (!argList->getSwitch("-style"))
-  {
-    if (WMHelper::isKDERunning())
-    {
-      //QApplication::setStyle(new QPlastiqueStyle());
-    }
-    else if (UnixCommand::getLinuxDistro() == ectn_MANJAROLINUX)
-    {
-      QApplication::setStyle(new QGtkStyle());
-    }
-    else
-    {
-      QCleanlooksStyle *cls = new QCleanlooksStyle();
-      cls->setObjectName("cleanlooks");
-      QApplication::setStyle(cls);
-    }
-  }
-  else
-  {
-    if (argList->contains("cleanlooks", Qt::CaseInsensitive))
-    {
-      QCleanlooksStyle *cls = new QCleanlooksStyle();
-      cls->setObjectName("cleanlooks");
-      QApplication::setStyle(cls);
-    }
-  }
-
   QtSingleApplication app( StrConstants::getApplicationName(), argc, argv );
 
   //This sends a message just to awake the socket-based QtSinleApplication engine
