--- src/mainwindow_transaction.cpp.orig	2013-04-06 23:03:57.000000000 -0400
+++ src/mainwindow_transaction.cpp	2013-04-07 14:04:30.208373260 -0400
@@ -541,7 +541,7 @@
                        this, SLOT( actionsProcessRaisedError() ));
 
       QString command;
-      command = "pacman -Su --noconfirm";
+      command = "konsole --nofork -e pacman -Su";
 
       m_lastCommandList.clear();
       m_lastCommandList.append("pacman -Su;");
@@ -651,8 +651,8 @@
                      this, SLOT( actionsProcessRaisedError() ));
 
     QString command;
-    command = "pacman -Rcs --noconfirm " + listOfRemoveTargets +
-        "; pacman -S --noconfirm " + listOfInstallTargets;
+    command = "konsole --nofork -e pacman -Rcs " + listOfRemoveTargets +
+        "; konsole --nofork -e pacman -S " + listOfInstallTargets;
 
     m_lastCommandList.clear();
     m_lastCommandList.append("pacman -Rcs " + listOfRemoveTargets + ";");
@@ -732,7 +732,7 @@
                      this, SLOT( actionsProcessRaisedError() ));
 
     QString command;
-    command = "pacman -Rcs --noconfirm " + listOfTargets;
+    command = "konsole --nofork -e pacman -Rcs " + listOfTargets;
 
     m_lastCommandList.clear();
     m_lastCommandList.append("pacman -Rcs " + listOfTargets + ";");
@@ -813,7 +813,7 @@
                      this, SLOT( actionsProcessRaisedError() ));
 
     QString command;
-    command = "pacman -S --noconfirm " + listOfTargets;
+    command = "konsole --nofork -e pacman -S " + listOfTargets;
 
     m_lastCommandList.clear();
     m_lastCommandList.append("pacman -S " + listOfTargets + ";");
