--- src/mainwindow_news.cpp.orig	2013-10-03 23:09:27.774349920 -0400
+++ src/mainwindow_news.cpp	2013-10-03 23:21:09.333801129 -0400
@@ -69,6 +69,11 @@
       writeToTabOutputExt("<b>" +
                           StrConstants::getSearchingForDistroNews().arg("Chakra") + "</b>");
     }
+    else if (gotoNewsTab && distro == ectn_KAOS)
+    {
+      writeToTabOutputExt("<b>" +
+                          StrConstants::getSearchingForDistroNews().arg("KaOS") + "</b>");
+    }
     else if (gotoNewsTab && distro == ectn_MANJAROLINUX)
     {
       writeToTabOutputExt("<b>" +
