--- src/mainwindow_news.cpp.orig	2013-05-12 17:48:10.468978332 -0400
+++ src/mainwindow_news.cpp	2013-05-12 17:48:27.072836149 -0400
@@ -43,6 +43,7 @@
 {
   const QString ctn_ARCH_LINUX_RSS = "https://www.archlinux.org/feeds/news/";
   const QString ctn_CHAKRA_RSS = "http://chakra-project.org/news/index.php?/feeds/index.rss2";
+  const QString ctn_KDEOS_RSS = "http://kde-os.tk/feed/";
   const QString ctn_MANJARO_LINUX_RSS = "http://manjaro.org/feed/";
 
   LinuxDistro distro = UnixCommand::getLinuxDistro();
@@ -73,6 +74,10 @@
     {
       curlCommand = curlCommand.arg(ctn_CHAKRA_RSS).arg(tmpRssPath);
     }
+    else if (distro == ectn_KDEOS)
+    {
+      curlCommand = curlCommand.arg(ctn_KDEOS_RSS).arg(tmpRssPath);
+    }
     else if (distro == ectn_MANJAROLINUX)
     {
       curlCommand = curlCommand.arg(ctn_MANJARO_LINUX_RSS).arg(tmpRssPath);
@@ -168,6 +173,10 @@
   {
     html = "<p align=\"center\"><h2>" + StrConstants::getChakraNews() + "</h2></p><ul>";
   }
+  else if (distro == ectn_KDEOS)
+  {
+    html = "<p align=\"center\"><h2>" + StrConstants::getKdeOSNews() + "</h2></p><ul>";
+  }
   else if (distro == ectn_MANJAROLINUX)
   {
     html = "<p align=\"center\"><h2>" + StrConstants::getManjaroLinuxNews() + "</h2></p><ul>";
@@ -287,6 +296,11 @@
       writeToTabOutputExt("<b>" +
                           StrConstants::getSearchingForDistroNews().arg("Chakra") + "</b>");
     }
+    else if (distro == ectn_KDEOS)
+    {
+      writeToTabOutputExt("<b>" +
+                          StrConstants::getSearchingForDistroNews().arg("KdeOS") + "</b>");
+    }
     else if (distro == ectn_MANJAROLINUX)
     {
       writeToTabOutputExt("<b>" +
