--- a/src/searchlineedit.cpp	2020-03-09 18:12:56.340896693 +0100
+++ b/src/searchlineedit.cpp	2020-03-09 18:12:56.245898768 +0100
@@ -219,10 +219,10 @@
     style += "padding-left: 20px;}";
     setStyleSheet(style);
 
-    /*QPalette palette(QApplication::palette());
+    QPalette palette(QApplication::palette());
     palette.setColor(QPalette::Base, Qt::lightGray);
-    palette.setColor(QPalette::Text, Qt::white);
-    setPalette(palette);*/
+    palette.setColor(QPalette::Text, Qt::darkRed);
+    setPalette(palette);
   }
 }
 
