--- src/strconstants.h.orig	2013-08-11 04:55:49.000000000 -0400
+++ src/strconstants.h	2013-08-13 22:50:58.708228155 -0400
@@ -25,7 +25,6 @@
 #include "wmhelper.h"
 #include "unixcommand.h"
 
-#include <QGtkStyle>
 #include <QApplication>
 
 class StrConstants{
@@ -51,7 +50,7 @@
   }
 
   static QString getForeignRepositoryName(){
-    return "AUR";
+    return "non-repo";
   }
 
   static QString getArchLinuxNews(){
@@ -61,6 +60,10 @@
   static QString getChakraNews(){
     return QObject::tr("Chakra news");
   }
+  
+  static QString getKaOSNews(){
+    return QObject::tr("KaOS news");
+  }
 
   static QString getManjaroLinuxNews(){
     return QObject::tr("Manjaro Linux news");
@@ -619,40 +622,6 @@
   static QString getTreeViewCSS(){
     QString res;    
 
-    if (qApp->style()->inherits("QGtkStyle") && UnixCommand::getLinuxDistro() == ectn_MANJAROLINUX && !WMHelper::isKDERunning())
-    {
-      /*QGtkStyle qgtkStyle;
-      QPalette palette = qgtkStyle.standardPalette();
-      QColor color = palette.color(QPalette::Highlight);*/
-
-      res = "QTreeView::branch:has-siblings:!adjoins-item {"
-                   "   border-image: url(:/resources/styles/vline.png) 0;}"
-                   "QTreeView::branch:has-siblings:adjoins-item {"
-                   "    border-image: url(:/resources/styles/branch-more.png) 0;}"
-                   "QTreeView::branch:!has-children:!has-siblings:adjoins-item {"
-                   "   border-image: url(:/resources/styles/branch-end.png) 0;}"
-                   "QTreeView::branch:has-children:!has-siblings:closed,"
-                   "QTreeView::branch:closed:has-children:has-siblings {"
-                   "       border-image: none;"
-                   "        image: url(:/resources/styles/branch-closed_BW.png);}"
-                   "QTreeView::branch:open:has-children:!has-siblings,"
-                   "QTreeView::branch:open:has-children:has-siblings  {"
-                   "       border-image: none;"
-                   "       image: url(:/resources/styles/branch-open_BW.png);}";
-
-                   /*"QTreeView {"
-          "selection-background-color: qlineargradient(x1: 0, y1: 0, x2: 0, y2: 1, stop: 0 " + color.name() + ", stop: 1 " + color.name() + ");"
-                   "show-decoration-selected: 1;"
-                   "}"
-                   "QTreeView::item:selected:active{"
-                   "background: qlineargradient(x1: 0, y1: 0, x2: 0, y2: 1, stop: 0 " + color.name() + ", stop: 1 " + color.name() + ");}"
-                   "QTreeView::item:selected:!active {"
-                   " background: qlineargradient(x1: 0, y1: 0, x2: 0, y2: 1, stop: 0 " + color.name() + ", stop: 1 " + color.name() + ");}"
-                   "QTreeView::item:selected {"
-                   "border: 1px solid " + color.name() +";"
-                   "}"*/
-    }
-    else
     {
       res = "QTreeView::branch:has-siblings:!adjoins-item {"
                      " border-image: url(:/resources/styles/vline.png) 0;}"
