--- src/strconstants.h.orig	2013-10-20 00:33:09.000000000 -0400
+++ src/strconstants.h	2013-10-21 16:22:55.646062336 -0400
@@ -24,7 +24,6 @@
 #include "wmhelper.h"
 #include "unixcommand.h"
 
-#include <QGtkStyle>
 #include <QApplication>
 
 class StrConstants{
@@ -50,7 +49,7 @@
   }
 
   static QString getForeignRepositoryName(){
-    return "AUR";
+    return "non-repo";
   }
 
   static QString getArchLinuxNews(){
@@ -61,6 +60,10 @@
     return QObject::tr("Chakra news");
   }
 
+  static QString getKaOSNews(){
+    return QObject::tr("KaOS news");
+  }
+  
   static QString getManjaroLinuxNews(){
     return QObject::tr("Manjaro Linux news");
   }
@@ -531,7 +534,7 @@
   static QString getTreeViewCSS(){
     QString res;    
 
-    //if (qApp->style()->inherits("QGtkStyle") && UnixCommand::getLinuxDistro() == ectn_MANJAROLINUX && !WMHelper::isKDERunning())
+    /*if (qApp->style()->inherits("QGtkStyle") && UnixCommand::getLinuxDistro() == ectn_MANJAROLINUX && !WMHelper::isKDERunning())
     //{
       res = "QTreeView::branch:has-siblings:!adjoins-item {"
                    "   border-image: url(:/resources/styles/vline.png) 0;}"
@@ -547,7 +550,7 @@
                    "QTreeView::branch:open:has-children:has-siblings  {"
                    "       border-image: none;"
                    "       image: url(:/resources/styles/branch-open_BW.png);}";
-    /*}
+    }
     else
     {
       res = "QTreeView::branch:has-siblings:!adjoins-item {"
