--- src/unixcommand.h.orig	2013-05-12 11:03:58.000000000 -0400
+++ src/unixcommand.h	2013-05-12 17:51:38.642187016 -0400
@@ -35,7 +35,7 @@
                         ectn_INSTALL, ectn_REMOVE, ectn_REMOVE_INSTALL,
                         ectn_RUN_IN_TERMINAL };
 
-enum LinuxDistro { ectn_ARCHLINUX, ectn_MANJAROLINUX, ectn_CHAKRA, ectn_UNKNOWN };
+enum LinuxDistro { ectn_ARCHLINUX, ectn_MANJAROLINUX, ectn_CHAKRA, ectn_KDEOS, ectn_UNKNOWN };
 
 //Forward class declarations.
 class QString;
