--- a/src/termwidget.cpp	2019-11-24 19:14:57.044210903 +0100
+++ b/src/termwidget.cpp	2019-11-24 19:14:56.954212828 +0100
@@ -39,10 +39,10 @@
   font.setFamily("Monospace");
   setTerminalFont(font);
   setScrollBarPosition(QTermWidget::ScrollBarRight);
-  changeDir("/");
+  //changeDir("/");
   setColorScheme("WhiteOnBlack");
-  execute("export TERM=xterm");
-  execute("clear");
+  //execute("export TERM=xterm");
+  //execute("clear");
   setContextMenuPolicy(Qt::CustomContextMenu);
 
   m_actionZoomIn = new QAction(this);
