--- a/src/widgets/freespacebar.cpp	2020-05-26 21:59:28.148873099 +0200
+++ b/src/widgets/freespacebar.cpp	2020-05-26 22:00:41.662284027 +0200
@@ -20,6 +20,7 @@
 
 #include <QLinearGradient>
 #include <QPainter>
+#include <QPainterPath>
 
 const int FreeSpaceBar::kBarHeight = 20;
 const int FreeSpaceBar::kBarBorderRadius = 8;

--- a/src/widgets/osdpretty.cpp	2020-05-26 22:14:46.379027869 +0200
+++ b/src/widgets/osdpretty.cpp	2020-05-26 22:20:25.972690095 +0200
@@ -26,6 +26,7 @@
 #include <QLayout>
 #include <QMouseEvent>
 #include <QPainter>
+#include <QPainterPath>
 #include <QSettings>
 #include <QTimer>
 #include <QTimeLine>

--- a/src/widgets/sliderwidget.cpp	2020-05-26 22:39:51.082630307 +0200
+++ b/src/widgetssliderwidget.cpp	2020-05-26 22:39:51.167628481 +0200
@@ -24,6 +24,7 @@
 #include <QBrush>
 #include <QImage>
 #include <QPainter>
+#include <QPainterPath>
 #include <QSize>
 #include <QTimer>
 #include <QStyle>
