--- a/src/modules/partition/gui/ChoicePage.cpp	2014-12-16 13:04:26.293294026 -0500
+++ b/src/modules/partition/gui/ChoicePage.cpp	2014-12-17 13:57:42.529000166 -0500
@@ -154,7 +154,7 @@
 
                 replaceButton->setText( tr( "<b>Install %1 on an existing partition</b><br/>"
                                             "<font color=\"red\">Warning: </font>This will delete all files "
-                                            "on the selected partition." )
+                                            "on the selected partition. <b>Not suitable for UEFI installs</b> " )
                                         .arg( Calamares::Branding::instance()->
                                               string( Calamares::Branding::VersionedName ) ) );
             )
@@ -180,7 +180,7 @@
 
                 replaceButton->setText( tr( "<b>Install %1 on an existing partition</b><br/>"
                                             "<font color=\"red\">Warning: </font>This will delete all files "
-                                            "on the selected partition." )
+                                            "on the selected partition. <b>Not suitable for UEFI installs</b> " )
                                         .arg( Calamares::Branding::instance()->
                                               string( Calamares::Branding::VersionedName ) ) );
             )
@@ -222,7 +222,7 @@
 
             replaceButton->setText( tr( "<b>Install %1 on an existing partition</b><br/>"
                                         "<font color=\"red\">Warning: </font>This will delete all files "
-                                        "on the selected partition." )
+                                        "on the selected partition. <b>Not suitable for UEFI installs</b>" )
                                     .arg( Calamares::Branding::instance()->
                                           string( Calamares::Branding::VersionedName ) ) );
         )
