--- a/src/calamares/CalamaresWindow.cpp	2014-08-15 18:39:38.565190849 -0400
+++ b/src/calamares/CalamaresWindow.cpp	2014-08-15 20:29:40.315313395 -0400
@@ -54,7 +54,10 @@
     QHBoxLayout* logoLayout = new QHBoxLayout;
     sideLayout->addLayout( logoLayout );
     logoLayout->addStretch();
-    QLabel* logoLabel = new QLabel( "branding\ngoes\nhere", sideBox );
+    QLabel* logoLabel = new QLabel( 0, sideBox );
+    QPixmap kaos(":/images/kaos.svgz");
+    logoLabel->setPixmap(kaos);
+    logoLabel->setMask(kaos.mask());
     {
         QPalette plt = sideBox->palette();
         sideBox->setAutoFillBackground( true );
