--- a/src/libcalamaresui/utils/CalamaresStyle.h	2014-08-15 18:39:38.565190849 -0400
+++ a/src/libcalamaresui/utils/CalamaresStyle.h	2014-08-15 18:44:10.602960805 -0400
@@ -26,8 +26,8 @@
 
 namespace CalamaresStyle {
 
-static const QColor SIDEBAR_BACKGROUND =  "#292F34";
-static const QColor SIDEBAR_TEXT =        "#FFFFFF";
+static const QColor SIDEBAR_BACKGROUND =  "#bdc3c7";
+static const QColor SIDEBAR_TEXT =        "#383838";
 
 } // namespace CalamaresStyle
 
