--- a/src/libcalamaresui/utils/CalamaresStyle.h	2014-10-27 22:08:57.898737422 -0400
+++ a/src/libcalamaresui/utils/CalamaresStyle.h	2014-10-28 21:52:37.142810998 -0400
@@ -26,9 +26,9 @@
 
 namespace CalamaresStyle {
 
-static const QColor SIDEBAR_BACKGROUND =  "#292F34";
-static const QColor SIDEBAR_TEXT =        "#FFFFFF";
-static const QColor SIDEBAR_TEXT_SELECT = "#292F34";
+static const QColor SIDEBAR_BACKGROUND =  "#bdc3c7";
+static const QColor SIDEBAR_TEXT =        "#1F1F1F";
+static const QColor SIDEBAR_TEXT_SELECT = "#3498DB";
 
 } // namespace CalamaresStyle
 