--- a/src/libcalamaresui/utils/CalamaresStyle.h	2014-08-15 18:39:38.565190849 -0400
+++ a/src/libcalamaresui/utils/CalamaresStyle.h	2014-10-18 17:46:07.135312946 -0400
@@ -26,8 +26,9 @@
 
 namespace CalamaresStyle {
 
-static const QColor SIDEBAR_BACKGROUND =  "#292F34";
-static const QColor SIDEBAR_TEXT =        "#FFFFFF";
+static const QColor SIDEBAR_BACKGROUND =  "#bdc3c7";
+static const QColor SIDEBAR_TEXT =        "#1F1F1F";
+static const QColor SIDEBAR_TEXT_SELECT = "#3498DB";
 
 } // namespace CalamaresStyle
 
