--- a/src/modules/partition/partitionmanager/calamares/CMakeLists.txt	2014-08-07 13:28:49.859638368 -0400
+++ b/src/modules/partition/partitionmanager/calamares/CMakeLists.txt	2014-08-07 13:28:50.186313988 -0400
@@ -19,7 +19,8 @@
     Config
     I18n
     CoreAddons
-    Solid)
+    Solid
+    ConfigWidgets)
 
 find_package(PkgConfig REQUIRED)
 pkg_check_modules(BLKID REQUIRED blkid)
@@ -129,6 +130,7 @@
         KF5::ConfigCore
         KF5::ConfigGui
         KF5::CoreAddons
+        KF5::ConfigWidgets
         KF5::I18n
         KF5::Solid
         libfatlabel

