--- a/src/modules/greeting/GreetingPage.cpp	2014-08-15 18:29:29.511131306 -0400
+++ b/src/modules/greeting/GreetingPage.cpp	2014-08-15 18:29:29.834473872 -0400
@@ -28,9 +28,10 @@
     QBoxLayout *mainLayout = new QHBoxLayout;
     setLayout( mainLayout );
 
-    QLabel* text = new QLabel( tr( "<h1>Welcome to Calamares.</h1><br/>"
-                                   "This is some random welcome text. "
-                                   "It should change depending on the branding config." ), this );
+    QLabel* text = new QLabel( tr( "<h1>Welcome to KaOS.</h1><br/>"
+                                   "This installer will guide you through the process of installing the KaOS live system<br/> "
+                                   "on your computer. After the installation you will have a ready-to-use KaOS system.<br/>"
+                                   "Remember, this installer is in alpha stage" ), this );
     text->setAlignment( Qt::AlignCenter );
 
     mainLayout->addStretch();
