--- a/src/modules/greeting/GreetingPage.cpp	2014-10-15 18:24:59.002223502 -0400
+++ a/src/modules/greeting/GreetingPage.cpp	2014-10-15 18:25:17.208473577 -0400
@@ -28,10 +28,15 @@
     QBoxLayout *mainLayout = new QHBoxLayout;
     setLayout( mainLayout );
 
-    QLabel* text = new QLabel( tr( "<h1>Welcome to Calamares.</h1><br/>"
-                                   "This is some random welcome text. "
-                                   "It should change depending on the branding config." ), this );
+    QLabel* text = new QLabel( tr( "<h1>Welcome to KaOS.</h1><br/>"
+                                    "This installer will guide you through the process of installing the KaOS live system on your computer.<br/> "
+                                    "After the installation you will have a ready-to-use KaOS system.<br/>"
+                                    "Remember, this installer is in alpha stage.<br/>"
+                                    "Not implemented yet is closing the installer automatically, branding with proper .svgz,<br/>"
+                                    "showing a slideshow during the install or setting the locale.<br/>" 
+                                    "For <b>GPT partioning on BIOS</b> systems and other instructions, read <a href=\"http://kaosx.us/download/\">Known Issues</a>" ), this );
     text->setAlignment( Qt::AlignCenter );
+    text->setOpenExternalLinks(true);
 
     mainLayout->addStretch();
     mainLayout->addWidget( text );
