--- a/src/modules/greeting/GreetingPage.cpp	2014-11-11 11:59:30.745224928 -0500
+++ b/src/modules/greeting/GreetingPage.cpp	2014-11-11 15:52:19.899637357 -0500
@@ -90,7 +90,11 @@
     CALAMARES_RETRANSLATE(
         m_text->setText( tr( "<h1>Welcome to the %1 installer.</h1><br/>"
                              "This program will ask you some questions and "
-                             "set up %2 on your computer." )
+                             "set up %2 on your computer." 
+                             " Remember, this installer is in Beta stage.<br/>"
+                             "Not implemented yet is closing the installer automatically,<br/>"
+                             "showing a slideshow during the install or setting the locale.<br/>" 
+                             "For <b>GPT partioning on BIOS</b> systems and other instructions, read <a href=\"http://kaosx.us/download/\">Known Issues</a>" )
                          .arg( Calamares::Branding::instance()->
                                string( Calamares::Branding::VersionedName ) )
                          .arg( Calamares::Branding::instance()->
