--- a/src/modules/partition/gui/PartitionLabelsView.cpp	2016-06-10 18:48:08.381024971 -0400
+++ b/src/modules/partition/gui/PartitionLabelsView.cpp	2016-06-10 18:53:28.736642959 -0400
@@ -161,7 +161,7 @@
         //HACK: horrible special casing follows.
         //      To save vertical space, we choose to hide short instances of free space.
         //      Arbitrary limit: 10MB.
-        const qint64 maxHiddenB = 10'000'000;
+        const qint64 maxHiddenB = qint64(10000000);
         if ( index.data( PartitionModel::IsFreeSpaceRole ).toBool() &&
              index.data( PartitionModel::SizeRole ).toLongLong() <  maxHiddenB )
             continue;
