--- a/src/calamares/progresstree/ProgressTreeDelegate.cpp	2014-08-24 22:44:29.860724727 -0400
+++ b/src/calamares/progresstree/ProgressTreeDelegate.cpp	2014-10-18 17:37:12.088891059 -0400
@@ -135,7 +135,7 @@
 
     if ( isCurrent )
     {
-        painter->setPen( CalamaresStyle::SIDEBAR_BACKGROUND );
+        painter->setPen( CalamaresStyle::SIDEBAR_TEXT_SELECT );
         painter->setBrush( APP->mainWindow()->palette().background() );
     }
 
