--- a/src/modules/partition/gui/ChoicePage.cpp	2014-11-11 16:08:37.359387598 -0500
+++ b/src/modules/partition/gui/ChoicePage.cpp	2014-11-11 16:11:40.184846150 -0500
@@ -109,7 +109,7 @@
 
             eraseButton->setText( tr( "<b>Erase disk and install %1</b><br/>"
                                       "<font color=\"red\">Warning: </font>This will delete all of your programs, "
-                                      "documents, photos, music, and any other files." )
+                                      "documents, photos, music, and any other files. <b>Not suitable for UEFI installs</b>" )
                                     .arg( Calamares::Branding::instance()->
                                           string( Calamares::Branding::VersionedName ) ) );
         )
@@ -130,7 +130,7 @@
                 alongsideButton->setText( tr( "<b>Install %2 alongside %1</b><br/>"
                                               "Documents, music and other personal files will be kept. "
                                               "You can choose which operating system you want each time the "
-                                              "computer starts up." )
+                                              "computer starts up.  <b>WARNING: not an option for UEFI installs</b>" )
                                             .arg( osName )
                                             .arg( Calamares::Branding::instance()->
                                                   string( Calamares::Branding::VersionedName ) ) );
@@ -138,7 +138,7 @@
                 eraseButton->setText( tr( "<b>Replace %1 with %2</b><br/>"
                                           "<font color=\"red\">Warning: </font>This will erase the whole disk and "
                                           "delete all of your %1 programs, "
-                                          "documents, photos, music, and any other files." )
+                                          "documents, photos, music, and any other files. <b>Not suitable for UEFI installs</b>" )
                                         .arg( osName )
                                         .arg( Calamares::Branding::instance()->
                                               string( Calamares::Branding::VersionedName ) ) );
@@ -153,13 +153,13 @@
                 alongsideButton->setText( tr( "<b>Install %1 alongside your current operating system</b><br/>"
                                               "Documents, music and other personal files will be kept. "
                                               "You can choose which operating system you want each time the "
-                                              "computer starts up." )
+                                              "computer starts up.  <b>WARNING: not an option for UEFI installs</b>" )
                                             .arg( Calamares::Branding::instance()->
                                                   string( Calamares::Branding::VersionedName ) ) );
 
                 eraseButton->setText( tr( "<b>Erase disk and install %1</b><br/>"
                                           "<font color=\"red\">Warning: </font>This will delete all of your Windows 7 programs, "
-                                          "documents, photos, music, and any other files." )
+                                          "documents, photos, music, and any other files. <b>Not suitable for UEFI installs</b>" )
                                         .arg( Calamares::Branding::instance()->
                                               string( Calamares::Branding::VersionedName ) ) );
             )
@@ -189,13 +189,13 @@
             alongsideButton->setText( tr( "<b>Install %1 alongside your current operating systems</b><br/>"
                                           "Documents, music and other personal files will be kept. "
                                           "You can choose which operating system you want each time the "
-                                          "computer starts up." )
+                                          "computer starts up.  <b>WARNING: not an option for UEFI installs</b>" )
                                         .arg( Calamares::Branding::instance()->
                                               string( Calamares::Branding::VersionedName ) ) );
 
             eraseButton->setText( tr( "<b>Erase disk and install %1</b><br/>"
                                       "<font color=\"red\">Warning: </font>This will delete all of your Windows 7 programs, "
-                                      "documents, photos, music, and any other files." )
+                                      "documents, photos, music, and any other files. <b>Not suitable for UEFI installs</b>" )
                                     .arg( Calamares::Branding::instance()->
                                           string( Calamares::Branding::VersionedName ) ) );
         )
@@ -212,7 +212,9 @@
     CALAMARES_RETRANSLATE(
         somethingElseButton->setText( tr( "<b>Something else</b><br/>"
                                           "You can create or resize partitions yourself, or choose "
-                                          "multiple partitions for %1." )
+                                          "multiple partitions for %1."
+                                          "  Use this option for UEFI installs, having a <b>fat32 512Mb /boot partition"
+                                          " is a must for UEFI installs</b>, either use an existing without formatting or create one." )
                                         .arg( Calamares::Branding::instance()->
                                               string( Calamares::Branding::ShortVersionedName ) ) );
     )
