diff --git a/app/main.cpp b/app/main.cpp
index 2afd684ca035595a9d18f8e6a6e8f017975fa170..b62c56bb7126d0ef25027f8e05b5f22f9375667e 100644
--- a/app/main.cpp
+++ b/app/main.cpp
@@ -51,6 +51,8 @@ int main(int argc, char *argv[])
     Tp::registerTypes();
     KTp::registerOtrTypes();
 
+    Tp::SharedPtr<TelepathyChatUi> app(new TelepathyChatUi(argc, argv));
+
     Tp::ChannelFactoryPtr channelFactory = Tp::ChannelFactory::create(QDBusConnection::sessionBus());
     channelFactory->addCommonFeatures(Tp::Channel::FeatureCore);
 
@@ -64,7 +66,6 @@ int main(int argc, char *argv[])
 
     Tp::ClientRegistrarPtr registrar = Tp::ClientRegistrar::create(KTp::accountFactory(), KTp::connectionFactory(), channelFactory, KTp::contactFactory());
 
-    Tp::SharedPtr<TelepathyChatUi> app(new TelepathyChatUi(argc, argv));
     Tp::AbstractClientPtr handler = Tp::AbstractClientPtr(app);
     registrar->registerClient(handler, QLatin1String(KTP_TEXTUI_CLIENT_NAME));
 
