Index: src/frontends/qt4/GuiViewSource.cpp
===================================================================
--- a/src/frontends/qt4/GuiViewSource.cpp	(revision dc5d056dd51a289483d996d5b21c55274044822d)
+++ b/src/frontends/qt4/GuiViewSource.cpp	(revision 6d375ddeac49ac8c6340398c9797354edca3f5d1)
@@ -219,12 +219,7 @@
 		// position-to-row is unavailable
 		// we jump to the first modification
-		const QChar * oc = old.constData();
-		const QChar * nc = qcontent.constData();
+		int length = min(old.length(), qcontent.length());
 		int pos = 0;
-		while (*oc != '\0' && *nc != '\0' && *oc == *nc) {
-			++oc;
-			++nc;
-			++pos;
-		}
+		for (; pos < length && old.at(pos) == qcontent.at(pos); ++pos) {}
 		QTextCursor c = QTextCursor(viewSourceTV->document());
 		//get some space below the cursor
