--- a/src/qmapshack/helpers/CDraw.cpp	2020-06-14 18:51:14.297307158 +0200
+++ b/src/qmapshack/helpers/CDraw.cpp	2020-06-14 18:51:14.297307158 +0200
@@ -24,6 +24,7 @@
 #include <QImage>
 #include <QPointF>
 #include <QtMath>
+#include <QPainterPath>
 
 QPen CDraw::penBorderBlue(QColor(10, 10, 150, 220), 2);
 QPen CDraw::penBorderGray(Qt::lightGray, 2);
--- a/src/qmaptool/helpers/CDraw.cpp	2020-06-14 18:51:14.297307158 +0200
+++ b/src/qmaptool/helpers/CDraw.cpp	2020-06-14 18:51:14.383305353 +0200
@@ -24,6 +24,7 @@
 #include <QImage>
 #include <QPointF>
 #include <QtMath>
+#include <QPainterPath>
 
 QPen CDraw::penBorderBlue(QColor(10,10,150,220),2);
 QPen CDraw::penBorderGray(Qt::lightGray,2);
