--- src/backends/x11.cmake_orig	2014-02-11 21:00:25.249027384 -0500
+++ src/backends/x11.cmake	2014-02-11 21:05:13.460055233 -0500
@@ -8,8 +8,8 @@
     message(FATAL_ERROR "Xinput not found")
 endif()
 
-pkg_search_module(SYNAPTICS REQUIRED xorg-synaptics)
-pkg_search_module(XORG REQUIRED xorg-server)
+#pkg_search_module(SYNAPTICS REQUIRED xorg-synaptics)
+#pkg_search_module(XORG REQUIRED xorg-server)
 
 include_directories(${XCB_INCLUDE_DIR}
                     ${X11_XCB_INCLUDE_DIR}
