--- a/plasma-applet-appmenu-qml/contents/config/main.xml     2013-02-07 15:37:00.000000000 -0500
+++ b/plasma-applet-appmenu-qml/contents/config/main.xml	2013-06-12 22:00:49.767119303 -0400
@@ -12,7 +12,7 @@
     </entry>
     <entry name="icon" type="String">
       <label>Icon on the launcher button</label>
-      <default>start-here-kde</default>
+      <default>/usr/share/icons/home.png</default>
     </entry>
     <entry name="favoritesVisible" type="Bool">
       <label>Show favorites list</label>
@@ -48,7 +48,7 @@
     </entry>
     <entry name="numOfColumns" type="Int">
       <label>Number of visible columns</label>
-      <default>2</default>
+      <default>3</default>
     </entry>
     <entry name="resetOnShow" type="Bool">
       <label>Reset menu each time it is shown</label>
@@ -76,7 +76,7 @@
     </entry>
     <entry name="suspendToRamButtonVisible" type="Bool">
       <label>Show "Sleep" button</label>
-      <default>false</default>
+      <default>true</default>
     </entry>
     <entry name="suspendToDiskButtonVisible" type="Bool">
       <label>Show "Hibernate" button</label>
--- b/plasma-applet-appmenu-qml/contents/ui/main.qml	2013-02-13 13:08:21.000000000 -0500
+++ b/plasma-applet-appmenu-qml/contents/ui/main.qml	2013-06-12 22:21:11.002536025 -0400
@@ -23,8 +23,8 @@
 
 Item {
 	id: main
-	property int minimumWidth: 150 // must set this in order for Plasma to "get" that this is really a PopupApplet
-	property int minimumHeight: 200 // idem
+	property int minimumWidth: 580 // must set this in order for Plasma to "get" that this is really a PopupApplet
+	property int minimumHeight: 580 // idem
 	property int maximumWidth
 	property int maximumHeight
 	property int preferredWidth: numOfColumns == 1 ? 250 : 500
@@ -62,14 +62,14 @@
 	}
 
 	// configurable variables (the defaults specified here should not have any effect because they are overwritten by the values in the config in configChanged())
-	property string popupIcon: "start-here-kde"
+	property string popupIcon: "/usr/share/icons/home.png"
 	property bool favoritesVisible: true
 	property bool favoritesLocked: false
 	property bool scrollOverBounds: true
 	property int iconSize: theme.smallMediumIconSize // 22
 	property int smallIconSize: theme.smallIconSize // 16
 	property bool showDescription: true
-	property int numOfColumns: 2 // number of visible columns in the widget
+	property int numOfColumns: 3 // number of visible columns in the widget
 	property bool resetOnShow: false
 	property bool checkEmptySubmenusThoroughly: false
 	// other variables

