--- a/src/templates/CMakeLists.txt
+++ b/src/templates/CMakeLists.txt
@@ -14,6 +14,5 @@
 install(FILES C++/runner/runner.png DESTINATION ${DATA_INSTALL_DIR}/kdevappwizard/template_previews)
 install(FILES C++/konqplugin/konqplugin.png DESTINATION ${DATA_INSTALL_DIR}/kdevappwizard/template_previews)
 install(FILES C++/cmake_qt5guiapp/cmake_qt5guiapp.png DESTINATION ${DATA_INSTALL_DIR}/kdevappwizard/template_previews)
-install(FILES QML/qml-plasmoid/qml-plasmoid.png DESTINATION ${DATA_INSTALL_DIR}/kdevappwizard/template_previews)
 install(FILES QML/qt5-qml2/qt5-qml2.png DESTINATION ${DATA_INSTALL_DIR}/kdevappwizard/template_previews)

--- a/src/templates/QML/CMakeLists.txt
+++ b/src/templates/QML/CMakeLists.txt
@@ -1,5 +1,4 @@
 set(apptemplate_DIRS
-    qml-plasmoid
     qt5-qml2)
 
 kapptemplate_add_app_templates(${apptemplate_DIRS})

