--- CMakeLists.txt.orig	2009-03-17 08:58:31.000000000 +0100
+++ CMakeLists.txt	2009-05-30 22:05:24.000000000 +0200
@@ -13,7 +13,7 @@
 
 ########### install files ###############
 
-set( oxygen_icon_dirs  8x8 16x16 22x22 32x32 48x48 64x64 128x128 256x256 )
+set( oxygen_icon_dirs  8x8 16x16 22x22 32x32 48x48 64x64 128x128 256x256 scalable)
 
 install( DIRECTORY ${oxygen_icon_dirs} DESTINATION ${ICON_INSTALL_DIR}/oxygen PATTERN .svn EXCLUDE PATTERN "*.sh" EXCLUDE )
 install( FILES index.theme DESTINATION ${ICON_INSTALL_DIR}/oxygen )
