--- a/shell/main.cpp	2016-04-29 18:18:57.595201352 -0400
+++ b/shell/main.cpp	2016-04-29 18:18:11.723341670 -0400
@@ -26,11 +26,37 @@
 #include "aboutdata.h"
 #include "okular_main.h"
 #include "shellutils.h"
+#include <QMessageBox>
+#include <QCheckBox>
 
 int main(int argc, char** argv)
 {
     QApplication app(argc, argv);
 
+    bool exit = false;
+     QSettings settings;
+     settings.beginGroup("MainWindow");
+     if (settings.value("quitWithoutPrompt").toBool()) {
+      exit = true;
+     } else {
+      QMessageBox quitQuestion;
+      QCheckBox checkBox;
+      checkBox.setText("Do not ask again");
+      quitQuestion.setCheckBox(&checkBox);
+      quitQuestion.setText("Do you understand KaOS ships an unreleased Frameworks 5 based version of Okular? Will you report bugs on the KaOS tracker and not in the offical KDE Bugtracker?");
+      quitQuestion.setWindowTitle("Okular Frameworks");
+      quitQuestion.setStandardButtons(QMessageBox::Ok);
+      quitQuestion.exec();
+      if (quitQuestion.result() == QMessageBox::Ok) {
+       settings.setValue("quitWithoutPrompt", checkBox.isChecked());
+       exit = true;
+      }
+     }
+     settings.endGroup();
+     if (exit) {
+      QCoreApplication::quit();
+     }
+
     KLocalizedString::setApplicationDomain("okular");
 
     KAboutData aboutData = okularAboutData();
