--- a/ImageLounge/src/DkCore/DkMetaData.cpp	2018-10-22 11:25:33.000000000 +0200
+++ b/ImageLounge/src/DkCore/DkMetaDatanew.cpp	2019-05-18 22:52:45.548371577 +0200
@@ -32,6 +32,8 @@
 #include "DkSettings.h"
 #include "DkTimer.h"
 
+#include <iostream>
+
 #pragma warning(push, 0)	// no warnings from includes - begin
 #include <QTranslator>
 #include <QObject>
--- a/ImageLounge/src/DkGui/DkNoMacs.cpp	2019-05-18 22:57:08.768523606 +0200
+++ b/ImageLounge/src/DkGui/DkNoMacs.cpp	2019-05-18 22:57:24.031184516 +0200
@@ -88,6 +88,8 @@
 #include <QApplication>
 #pragma warning(pop)		// no warnings from includes - end
 
+#include <iostream>
+
 #if defined(Q_OS_WIN) && !defined(SOCK_STREAM)
 #include <winsock2.h>	// needed since libraw 0.16
 #endif
