--- a/src/core/symbology/qgsarrowsymbollayer.cpp	2020-05-28 11:46:14.282298183 +0200
+++ b/src/core/symbology/qgsarrowsymbollayer.cpp	2020-05-28 11:46:14.368296753 +0200
@@ -15,6 +15,7 @@
 
 #include "qgsarrowsymbollayer.h"
 #include "qgssymbollayerutils.h"
+#include <QPainterPath>
 
 QgsArrowSymbolLayer::QgsArrowSymbolLayer()
 {

--- a/src/core/symbology/qgsfillsymbollayer.cpp	2020-05-28 11:53:43.568173964 +0200
+++ b/src/core/symbology/qgsfillsymbollayer.cpp	2020-05-28 11:53:43.686171734 +0200
@@ -35,6 +35,7 @@
 #include "qgslinestring.h"
 
 #include <QPainter>
+#include <QPainterPath>
 #include <QFile>
 #include <QSvgRenderer>
 #include <QDomDocument>

--- a/src/core/symbology/qgsmarkersymbollayer.cpp	2020-05-28 14:41:37.702099242 +0200
+++ b/src/core/symbology/qgsmarkersymbollayer.cpp	2020-05-28 14:41:37.821096813 +0200
@@ -27,6 +27,7 @@
 #include "qgsunittypes.h"
 
 #include <QPainter>
+#include <QPainterPath>
 #include <QSvgRenderer>
 #include <QFileInfo>
 #include <QDir>
