--- a/telegram-qt/telegram-qt/CTelegramDispatcher.hpp	2015-07-23 14:08:15.000000000 -0400
+++ b/telegram-qt/telegram-qt/CTelegramDispatcher.hpp	2015-07-23 14:12:23.895245546 -0400
@@ -25,6 +25,7 @@
 #include <QPair>
 #include <QStringList>
 #include <QVector>
+#include <QIODevice>
 
 #include "TLTypes.hpp"
 #include "TelegramNamespace.hpp"
--- a/telegram-qt/testApp/MainWindow.cpp	2015-07-23 14:17:08.000000000 -0400
+++ b/telegram-qt/testApp/MainWindow.cpp	2015-07-23 14:25:08.553095894 -0400
@@ -27,6 +27,7 @@
 #include <QCompleter>
 #include <QToolTip>
 #include <QStringListModel>
+#include <QObject>
 
 #include <QDebug>
 
