--- a/src/main.cpp	2013-02-02 18:49:59.000000000 -0500
+++ b/src/main.cpp	2013-02-03 15:29:35.158370124 -0500
@@ -51,8 +51,8 @@
     }
 
     // Fall back to gconf
-    if (themeName.isEmpty())
-        themeName = QGtkStyle::getGConfString(QLatin1String("/desktop/gnome/interface/gtk_theme"));
+    //if (themeName.isEmpty())
+    //   themeName = QGtkStyle::getGConfString(QLatin1String("/desktop/gnome/interface/gtk_theme"));
 
     return themeName;
 }
