diff --git a/komparemodellist.cpp b/komparemodellist.cpp
index 32242777411ddb8549154a6f2ef0fbc9fff7a239..dded4ebfc93c0ca312df2514cb28a5ff8413740c 100644
--- a/komparemodellist.cpp
+++ b/komparemodellist.cpp
@@ -417,14 +417,14 @@ bool KompareModelList::saveDestination( DiffModel* model )
 				return false;
 			}
 		}
-		KIO::FileCopyJob* copyJob = KIO::file_copy( QUrl::fromLocalFile( temp.fileName() ), fullDestinationPath );
+		KIO::FileCopyJob* copyJob = KIO::file_copy( QUrl::fromLocalFile( temp.fileName() ), fullDestinationPath, -1, KIO::Overwrite);
 		result = copyJob->exec();
 	}
 	else
 	{
 		qCDebug(LIBKOMPAREDIFF2) << "Tempfilename   : " << temp.fileName();
 		qCDebug(LIBKOMPAREDIFF2) << "DestinationURL : " << m_info->destination;
-		KIO::FileCopyJob* copyJob = KIO::file_copy( QUrl::fromLocalFile( temp.fileName() ), m_info->destination );
+		KIO::FileCopyJob* copyJob = KIO::file_copy( QUrl::fromLocalFile( temp.fileName() ), m_info->destination , -1, KIO::Overwrite);
 		result = copyJob->exec();
 		qCDebug(LIBKOMPAREDIFF2) << "true or false?" << result;
 	}
