diff -Npur partitionmanager-svn.orig/src/gui/createpartitiontablewidgetbase.ui partitionmanager-svn/src/gui/createpartitiontablewidgetbase.ui
--- partitionmanager-svn.orig/src/gui/createpartitiontablewidgetbase.ui	2011-03-10 12:02:58.000000000 +0100
+++ partitionmanager-svn/src/gui/createpartitiontablewidgetbase.ui	2011-03-10 12:05:46.236666677 +0100
@@ -6,7 +6,7 @@
    <rect>
     <x>0</x>
     <y>0</y>
-    <width>325</width>
+    <width>348</width>
     <height>193</height>
    </rect>
   </property>
@@ -25,11 +25,14 @@
     <layout class="QVBoxLayout" name="verticalLayout" stretch="0,0">
      <item>
       <widget class="QRadioButton" name="m_RadioGPT">
+       <property name="enabled">
+        <bool>false</bool>
+       </property>
        <property name="text">
         <string>GPT</string>
        </property>
        <property name="checked">
-        <bool>true</bool>
+        <bool>false</bool>
        </property>
       </widget>
      </item>
@@ -39,7 +42,7 @@
         <string>MS-Dos</string>
        </property>
        <property name="checked">
-        <bool>false</bool>
+        <bool>true</bool>
        </property>
       </widget>
      </item>
