--- a/CMakeLists.txt	2019-02-09 09:52:35.000000000 +0100
+++ b/CMakeLists.txt	2019-02-12 12:35:59.249595710 +0100
@@ -126,10 +126,12 @@
 find_package( Qt5Widgets REQUIRED )
 find_package( Qt5Core REQUIRED )
 find_package( Qt5Network REQUIRED )
+find_package( Qt5DBus REQUIRED )
 
 message( STATUS "Found Qt5Widgets, version  ${Qt5Widgets_VERSION}" )
 message( STATUS "Found Qt5Core, version  ${Qt5Core_VERSION}" )
 message( STATUS "Found Qt5Network, version  ${Qt5Network_VERSION}" )
+message( STATUS "Found Qt5DBus, version  ${Qt5DBus_VERSION}" )
 
 file( WRITE ${PROJECT_BINARY_DIR}/version.h "
 #ifndef THIS_VERSION
