--- a/src/activities/menu/Menu.qml	2016-04-03 16:09:41.138612421 -0400
+++ b/src/activities/menu/Menu.qml	2016-04-03 16:09:40.432637393 -0400
@@ -410,8 +410,6 @@
                     ActivityInfoTree.currentActivity = ActivityInfoTree.menuTree[index]
                     activityLoader.setSource("qrc:/gcompris/src/activities/" + ActivityInfoTree.menuTree[index].name,
                                              {
-                                                 'audioVoices': audioVoices,
-                                                 'audioEffects': audioEffects,
                                                  'loading': loading,
                                                  'menu': menuActivity,
                                                  'activityInfo': ActivityInfoTree.currentActivity
