diff -Naur kdelibs-4.3.98.orig/kded/applications.menu kdelibs-4.3.98/kded/applications.menu
--- kdelibs-4.3.98.orig/kded/applications.menu	2010-01-31 20:28:11.000000000 +0100
+++ kdelibs-4.3.98/kded/applications.menu	2010-02-01 14:30:58.000000000 +0100
@@ -373,6 +373,13 @@
 		</Include>
 	</Menu>
 	<Menu>
+		<Name>Chakra</Name>
+		<Directory>Chakra.directory</Directory>
+		<Include>
+			<Category>Chakra</Category>
+		</Include>
+	</Menu>
+	<Menu>
 		<Name>System</Name>
 		<Directory>kde-system.directory</Directory>
 		<Include>