diff -Naur kdelibs-4.9.1-orig/kded/applications.menu kdelibs-4.9.1/kded/applications.menu
--- kdelibs-4.9.1-orig/kded/applications.menu	2012-08-13 10:39:49.000000000 +0200
+++ kdelibs-4.9.1/kded/applications.menu	2012-09-29 17:50:31.554171400 +0200
@@ -437,5 +437,6 @@
 		</And>
 	</Include>
 	<DefaultMergeDirs/>
+	<MergeDir>applications-merged</MergeDir>
 	<MergeFile>applications-kmenuedit.menu</MergeFile>
 </Menu>
