--- a/tools/svg2pnt/svgxmlhandler.cpp	2015-06-08 16:06:24.000000000 -0400
+++ b/tools/svg2pnt/svgxmlhandler.cpp	2015-06-27 13:44:33.136234102 -0400
@@ -12,6 +12,7 @@
 
 #include "svgxmlhandler.h"
 #include <QDebug>
+#include <QDataStream>
 
 SVGXmlHandler::SVGXmlHandler(const QString& targetfile)
 {
--- a/tools/stars/stars.cpp	2015-06-08 16:06:24.000000000 -0400
+++ b/tools/stars/stars.cpp	2015-06-27 13:56:30.641500949 -0400
@@ -16,6 +16,7 @@
 #include <QApplication>
 #include <QStringList>
 #include <QFileInfo>
+#include <QDataStream>
 
 #include <cmath>
 #include <iostream>
--- a/src/lib/marble/geodata/data/GeoDataGeometry.cpp	2015-06-08 16:06:24.000000000 -0400
+++ b/src/lib/marble/geodata/data/GeoDataGeometry.cpp	2015-06-27 14:05:22.904875911 -0400
@@ -21,6 +21,8 @@
 
 #include "MarbleDebug.h"
 
+#include <QDataStream>
+
 
 namespace Marble
 {
--- a/src/lib/marble/geodata/data/GeoDataLineStyle.cpp	2015-06-08 16:06:24.000000000 -0400
+++ b/src/lib/marble/geodata/data/GeoDataLineStyle.cpp	2015-06-27 14:11:30.519482250 -0400
@@ -13,6 +13,8 @@
 
 #include "GeoDataTypes.h"
 
+#include <QDataStream>
+
 namespace Marble
 {
 
--- a/src/lib/marble/geodata/data/GeoDataMultiTrack.cpp	2015-06-08 16:06:24.000000000 -0400
+++ b/src/lib/marble/geodata/data/GeoDataMultiTrack.cpp	2015-06-27 14:18:45.352067036 -0400
@@ -18,6 +18,8 @@
 
 #include "MarbleDebug.h"
 
+#include <QDataStream>
+
 
 namespace Marble
 {
 
--- a/src/lib/marble/geodata/data/GeoDataLatLonAltBox.cpp	2015-06-27 14:32:30.000000000 -0400
+++ b/src/lib/marble/geodata/data/GeoDataLatLonAltBox.cpp	2015-06-27 14:32:42.000000000 -0400
@@ -18,6 +18,8 @@
 
 #include "GeoDataTypes.h"
 
+#include <QDataStream>
+
 namespace Marble
 {
 
--- a/src/lib/marble/geodata/data/GeoDataPolygon.cpp	2015-06-08 16:06:24.000000000 -0400
+++ b/src/lib/marble/geodata/data/GeoDataPolygon.cpp	2015-06-27 14:40:29.389867841 -0400
@@ -15,6 +15,8 @@
 
 #include "MarbleDebug.h"
 
+#include <QDataStream>
+
 
 namespace Marble
 {
--- a/src/lib/marble/geodata/data/GeoDataLineString.cpp	2015-06-08 16:06:24.000000000 -0400
+++ b/src/lib/marble/geodata/data/GeoDataLineString.cpp	2015-06-27 14:46:43.890218961 -0400
@@ -18,6 +18,8 @@
 #include "Quaternion.h"
 #include "MarbleDebug.h"
 
+#include <QDataStream>
+
 
 namespace Marble
 {
--- a/src/lib/marble/geodata/data/GeoDataPolyStyle.cpp	2015-06-08 16:06:24.000000000 -0400
+++ b/src/lib/marble/geodata/data/GeoDataPolyStyle.cpp	2015-06-27 14:51:03.126768662 -0400
@@ -13,6 +13,8 @@
 
 #include "GeoDataTypes.h"
 
+#include <QDataStream>
+
 namespace Marble
 {
 
--- a/src/lib/marble/geodata/data/GeoDataLabelStyle.cpp	2015-06-08 16:06:24.000000000 -0400
+++ b/src/lib/marble/geodata/data/GeoDataLabelStyle.cpp	2015-06-27 14:54:59.470713034 -0400
@@ -12,6 +12,7 @@
 #include "GeoDataLabelStyle.h"
 
 #include <QFont>
+#include <QDataStream>
 
 #include "GeoDataTypes.h"
 
--- a/src/lib/marble/geodata/data/GeoDataMultiGeometry.cpp	2015-06-08 16:06:24.000000000 -0400
+++ b/src/lib/marble/geodata/data/GeoDataMultiGeometry.cpp	2015-06-27 14:59:11.267610897 -0400
@@ -19,6 +19,8 @@
 
 #include "MarbleDebug.h"
 
+#include <QDataStream>
+
 
 namespace Marble
 {
--- a/src/lib/marble/geodata/data/GeoDataCoordinates.cpp	2015-06-08 16:06:24.000000000 -0400
+++ b/src/lib/marble/geodata/data/GeoDataCoordinates.cpp	2015-06-27 15:02:59.381754986 -0400
@@ -23,6 +23,7 @@
 #include <QStringList>
 #include <QCoreApplication>
 #include <QAtomicInt>
+#include <QDataStream>
 
 #include "MarbleGlobal.h"
 #include "MarbleDebug.h"
--- a/src/lib/marble/geodata/data/GeoDataLatLonBox.cpp	2015-06-08 16:06:24.000000000 -0400
+++ b/src/lib/marble/geodata/data/GeoDataLatLonBox.cpp	2015-06-27 15:08:58.957209673 -0400
@@ -18,6 +18,8 @@
 
 #include "GeoDataTypes.h"
 
+#include <QDataStream>
+
 namespace Marble
 {
 
--- a/src/lib/marble/DiscCache.cpp	2015-06-08 16:06:24.000000000 -0400
+++ b/src/lib/marble/DiscCache.cpp	2015-06-27 15:19:14.765670318 -0400
@@ -16,6 +16,7 @@
 #include <QtGlobal>
 #include <QFile>
 #include <QDirIterator>
+#include <QDataStream>
 
 using namespace Marble;
 
---  a/src/lib/marble/routing/RoutingModel.h	2015-06-08 16:06:24.000000000 -0400
+++  b/src/lib/marble/routing/RoutingModel.h	2015-06-27 15:37:13.608006460 -0400
@@ -13,6 +13,7 @@
 
 #include "marble_export.h"
 #include <QAbstractListModel>
+#include <QIODevice>
 
 #include "GeoDataCoordinates.h"
 
--- a/src/plugins/render/satellites/SatellitesConfigAbstractItem.h	2015-06-08 16:06:24.000000000 -0400
+++ b/src/plugins/render/satellites/SatellitesConfigAbstractItem.h	2015-06-27 15:57:25.390520939 -0400
@@ -13,6 +13,8 @@
 
 #include <QVector>
 #include <QStringList>
+#include <QHash>
+#include <QVariant>
 
 namespace Marble
 {
--- a/src/plugins/runner/cache/CacheRunner.cpp	2015-06-08 16:06:24.000000000 -0400
+++ b/src/plugins/runner/cache/CacheRunner.cpp	2015-06-27 16:52:21.243813051 -0400
@@ -14,6 +14,7 @@
 #include "GeoDataPlacemark.h"
 
 #include <QFile>
+#include <QDataStream>
 
 namespace Marble
 {
--- a/src/plugins/runner/json/JsonParser.h	2015-06-08 16:06:24.000000000 -0400
+++ b/src/plugins/runner/json/JsonParser.h	2015-06-27 16:58:25.291853448 -0400
@@ -16,6 +16,7 @@
 #include <QScriptEngine>
 #include <QScriptValue>
 #include <QScriptValueIterator>
+#include <QIODevice>
 
 namespace Marble {
 
--- a/src/plugins/runner/pn2/Pn2Runner.h	2015-06-08 16:06:24.000000000 -0400
+++ b/src/plugins/runner/pn2/Pn2Runner.h	2015-06-27 17:04:35.920441098 -0400
@@ -12,6 +12,8 @@
 
 #include "ParsingRunner.h"
 
+#include <QDataStream>
+
 class QDataStream;
 
 namespace Marble

--- a/src/plugins/runner/pnt/PntRunner.cpp	2015-06-08 16:06:24.000000000 -0400
+++ b/src/plugins/runner/pnt/PntRunner.cpp	2015-06-27 17:09:50.574680146 -0400
@@ -19,6 +19,7 @@
 #include <qmath.h>
 #include <QFile>
 #include <QFileInfo>
+#include <QDataStream>
 
 namespace Marble
 {
--- a/tools/dateline/dateline.cpp	2015-06-08 16:06:24.000000000 -0400
+++ b/tools/dateline/dateline.cpp	2015-06-27 17:14:30.185577047 -0400
@@ -17,6 +17,7 @@
 #include <QFile>
 #include <QVector>
 #include <QStringList>
+#include <QDataStream>
 #include "Quaternion.h"
 
 using namespace Marble;
--- a/tools/kml2kml/kml2kml.cpp	2015-06-08 16:06:24.000000000 -0400
+++ b/tools/kml2kml/kml2kml.cpp	2015-06-27 17:20:20.414548707 -0400
@@ -19,6 +19,7 @@
 #include <QApplication>
 #include <QFile>
 #include <QDebug>
+#include <QDataStream>
 
 using namespace std;
 using namespace Marble;
--- /tools/kml2cache/kml2cache.cpp	2015-06-08 16:06:24.000000000 -0400
+++ /tools/kml2cache/kml2cache.cpp	2015-06-27 17:34:55.704447999 -0400
@@ -24,6 +24,7 @@
 #include <QDebug>
 #include <QFile>
 #include <iostream>
+#include <QDataStream>
 
 using namespace std;
 using namespace Marble;
