diff --git a/src/translations/qjackctl_cs.ts b/src/translations/qjackctl_cs.ts
index 2fb6a59..3ff8d6e 100644
--- a/src/translations/qjackctl_cs.ts
+++ b/src/translations/qjackctl_cs.ts
@@ -29,7 +29,7 @@
     <message>
         <location filename="../qjackctlSetup.cpp" line="46"/>
         <source>(default)</source>
-        <translation>(výchozí)</translation>
+        <translation></translation>
     </message>
     <message>
         <location filename="../qjackctlSetup.cpp" line="492"/>
@@ -2247,7 +2247,7 @@ Jste si jistý, že chcete přepsat stávající složku?</translation>
         <location filename="../qjackctlSetupForm.ui" line="1783"/>
         <location filename="../qjackctlSetupForm.ui" line="3963"/>
         <source>(default)</source>
-        <translation>(výchozí)</translation>
+        <translation></translation>
     </message>
     <message>
         <location filename="../qjackctlSetupForm.ui" line="101"/>
diff --git a/src/translations/qjackctl_de.ts b/src/translations/qjackctl_de.ts
index ec70c81..ac61493 100644
--- a/src/translations/qjackctl_de.ts
+++ b/src/translations/qjackctl_de.ts
@@ -29,7 +29,7 @@
     <message>
         <location filename="../qjackctlSetup.cpp" line="46"/>
         <source>(default)</source>
-        <translation>(voreingestellt)</translation>
+        <translation></translation>
     </message>
     <message>
         <location filename="../qjackctlSetup.cpp" line="492"/>
@@ -2206,7 +2206,7 @@ Soll das existierende Verzeichnis überschrieben werden?</translation>
         <location filename="../qjackctlSetupForm.ui" line="1783"/>
         <location filename="../qjackctlSetupForm.ui" line="3963"/>
         <source>(default)</source>
-        <translation>(voreinst.)</translation>
+        <translation></translation>
     </message>
     <message>
         <location filename="../qjackctlSetupForm.ui" line="101"/>
diff --git a/src/translations/qjackctl_fr.ts b/src/translations/qjackctl_fr.ts
index 4cee09a..619d4e9 100644
--- a/src/translations/qjackctl_fr.ts
+++ b/src/translations/qjackctl_fr.ts
@@ -29,7 +29,7 @@
     <message>
         <location filename="../qjackctlSetup.cpp" line="46"/>
         <source>(default)</source>
-        <translation>(par défaut)</translation>
+        <translation></translation>
     </message>
     <message>
         <location filename="../qjackctlSetup.cpp" line="497"/>
@@ -2241,7 +2241,7 @@ Are you sure to overwrite the existing folder?</source>
         <location filename="../qjackctlSetupForm.ui" line="1783"/>
         <location filename="../qjackctlSetupForm.ui" line="3963"/>
         <source>(default)</source>
-        <translation>(par défaut)</translation>
+        <translation></translation>
     </message>
     <message>
         <location filename="../qjackctlSetupForm.ui" line="86"/>
diff --git a/src/translations/qjackctl_it.ts b/src/translations/qjackctl_it.ts
index 210935b..42aed8f 100644
--- a/src/translations/qjackctl_it.ts
+++ b/src/translations/qjackctl_it.ts
@@ -29,7 +29,7 @@
     <message>
         <location filename="../qjackctlSetup.cpp" line="46"/>
         <source>(default)</source>
-        <translation>(predefinito)</translation>
+        <translation></translation>
     </message>
     <message>
         <location filename="../qjackctlSetup.cpp" line="497"/>
@@ -2251,7 +2251,7 @@ Sovrascrivo la cartella?</translation>
         <location filename="../qjackctlSetupForm.ui" line="1783"/>
         <location filename="../qjackctlSetupForm.ui" line="3963"/>
         <source>(default)</source>
-        <translation>(predefinito)</translation>
+        <translation></translation>
     </message>
     <message>
         <location filename="../qjackctlSetupForm.ui" line="86"/>
diff --git a/src/translations/qjackctl_ja.ts b/src/translations/qjackctl_ja.ts
index 1ee71e4..8a48d5c 100644
--- a/src/translations/qjackctl_ja.ts
+++ b/src/translations/qjackctl_ja.ts
@@ -29,7 +29,7 @@
     <message>
         <location filename="../qjackctlSetup.cpp" line="46"/>
         <source>(default)</source>
-        <translation type="unfinished">(既定)</translation>
+        <translation></translation>
     </message>
     <message>
         <location filename="../qjackctlSetup.cpp" line="492"/>
diff --git a/src/translations/qjackctl_nl.ts b/src/translations/qjackctl_nl.ts
index 12b5bd0..0065c71 100644
--- a/src/translations/qjackctl_nl.ts
+++ b/src/translations/qjackctl_nl.ts
@@ -29,7 +29,7 @@
     <message>
         <location filename="../qjackctlSetup.cpp" line="46"/>
         <source>(default)</source>
-        <translation>(standaard)</translation>
+        <translation></translation>
     </message>
     <message>
         <location filename="../qjackctlSetup.cpp" line="497"/>
@@ -2254,7 +2254,7 @@ Bent u zeker dat u de bestaande map wil vervangen ?</translation>
         <location filename="../qjackctlSetupForm.ui" line="1783"/>
         <location filename="../qjackctlSetupForm.ui" line="3963"/>
         <source>(default)</source>
-        <translation>(standaard)</translation>
+        <translation></translation>
     </message>
     <message>
         <location filename="../qjackctlSetupForm.ui" line="86"/>
diff --git a/src/translations/qjackctl_ru.ts b/src/translations/qjackctl_ru.ts
index e025760..c9a5a1d 100644
--- a/src/translations/qjackctl_ru.ts
+++ b/src/translations/qjackctl_ru.ts
@@ -49,7 +49,7 @@
     <message>
         <location filename="../qjackctlSetup.cpp" line="46"/>
         <source>(default)</source>
-        <translation>(как обычно)</translation>
+        <translation></translation>
     </message>
     <message>
         <location filename="../qjackctlSetup.cpp" line="492"/>
@@ -2520,7 +2520,7 @@ Are you sure to overwrite the existing folder?</source>
         <location filename="../qjackctlSetupForm.ui" line="1783"/>
         <location filename="../qjackctlSetupForm.ui" line="3963"/>
         <source>(default)</source>
-        <translation>(как обычно)</translation>
+        <translation></translation>
     </message>
     <message>
         <location filename="../qjackctlSetupForm.ui" line="644"/>
