--- a/install.py	2021-05-02 20:33:53.526168422 +0200
+++ b/install.py	2021-05-02 20:33:53.620169171 +0200
@@ -1628,9 +1628,9 @@
             minor = int(minor)
             pat = int(pat)
             if (
-                major < 5 or
-                (major == 5 and minor < 0) or
-                (major == 5 and minor == 0 and pat < 0)
+                major < 4 or
+                (major == 4 and minor < 14) or
+                (major == 4 and minor == 14 and pat < 2)
             ):
                 print('Sorry, you must have sip 5.0.0 or higher or'
                       ' a recent snapshot release.')
