--- a/skgbasemodeler/skgdocument.cpp	2019-02-11 21:35:17.000000000 +0100
+++ b/skgbasemodeler/skgdocument.cpp	2019-03-16 11:13:43.306249380 +0100
@@ -1523,7 +1523,7 @@
                      << QStringLiteral("PRAGMA journal_mode=MEMORY")
                      << QStringLiteral("PRAGMA temp_store=MEMORY")
                      // << QStringLiteral("PRAGMA locking_mode=EXCLUSIVE")
-                     << QStringLiteral("PRAGMA synchronous = OFF")
+                      << QStringLiteral("PRAGMA legacy_alter_table=ON")  // For migration on sqlite >=3.25 (see https://sqlite.org/lang_altertable.html)
                      << QStringLiteral("PRAGMA recursive_triggers=true");
         IFOKDO(err, executeSqliteOrders(optimization))
