--- a/calendarsupport/CMakeLists.txt	2015-07-06 22:11:14.000000000 -0400
+++ b/calendarsupport/CMakeLists.txt	2015-07-06 23:06:46.013688031 -0400
@@ -90,10 +92,11 @@
 generate_export_header(calendarsupportcollectionpage BASE_NAME calendarsupportcollectionpage)
 target_link_libraries(
   calendarsupportcollectionpage
+PUBLIC
+  mailcommon
 PRIVATE
   KF5::AkonadiCore
   KF5::CalendarCore
-  mailcommon
   pimcommon
   KF5::AkonadiCalendar
 )
