--- a/applets/notes/package/contents/ui/main.qml	2016-08-03 18:15:23.578582014 -0400
+++ b/applets/notes/package/contents/ui/main.qml	2016-08-03 18:15:23.012607372 -0400
@@ -41,8 +41,8 @@
     }
     elementId: plasmoid.configuration.color + "-notes"
 
-    width: units.gridUnit * 15
-    height: units.gridUnit * 15
+    width: units.gridUnit * 35
+    height: units.gridUnit * 24
     Layout.minimumWidth: units.iconSizes.medium
     Layout.minimumHeight: units.iconSizes.medium
     Plasmoid.switchWidth: units.gridUnit * 5
