--- a/source/i18n/L10n.cpp.orig	2020-11-30 15:49:50.701955360 -0500
+++ b/source/i18n/L10n.cpp	2020-11-30 15:50:01.953176524 -0500
@@ -205,7 +205,7 @@ void L10n::ReevaluateCurrentLocaleAndRel
 	else
 	{
 		GetDictionaryLocale(locale, currentLocale);
-		currentLocaleIsOriginalGameLocale = (currentLocale == icu::Locale::getUS()) == TRUE;
+		currentLocaleIsOriginalGameLocale = (currentLocale == icu::Locale::getUS()) == true;
 		useLongStrings = false;
 	}
 	LoadDictionaryForCurrentLocale();
