--- a/notifier/mainwindow.cpp	2022-02-05 14:05:30.852202552 +0100
+++ b/notifier/mainwindow1.cpp	2022-02-05 14:05:30.918203098 +0100
@@ -562,10 +562,10 @@
     {
       if (m_transactionDialog->isMinimized())
         m_transactionDialog->setWindowState(Qt::WindowNoState);
-      else
+      /*else
       {
         m_transactionDialog->activateWindow();
-      }
+      }*/
 
       return;
     }
@@ -1127,7 +1127,7 @@
       if (m_outputDialog->isMinimized())
         m_outputDialog->setWindowState(Qt::WindowNoState);
 
-      m_outputDialog->activateWindow();
+      //m_outputDialog->activateWindow();
     }
     return;
   }
@@ -1191,7 +1191,7 @@
       if (m_outputDialog->isMinimized())
         m_outputDialog->setWindowState(Qt::WindowNoState);
 
-      m_outputDialog->activateWindow();
+      //m_outputDialog->activateWindow();
     }
     return;
   }
@@ -1306,6 +1306,6 @@
   }
   else
   {
-    m_optionsDialog->activateWindow();
+    //m_optionsDialog->activateWindow();
   }
 }
