--- a/src/QtSolutions/qtsingleapplication.cpp	2022-02-05 14:01:55.946426729 +0100
+++ b/src/QtSolutions/qtsingleapplication.cpp	2022-02-05 14:01:56.010427268 +0100
@@ -323,8 +323,8 @@
         actWin->raise();
         if (actWin->isHidden())
           actWin->show();
-        else
-          actWin->activateWindow();
+        //else
+          //actWin->activateWindow();
     }
   else if (actWin && message == QLatin1String("HIDE")) {
     if (!actWin->isHidden())
@@ -362,8 +362,8 @@
     actWin->raise();
     if (actWin->isHidden())
       actWin->show();
-    else
-      actWin->activateWindow();
+    //else
+      //actWin->activateWindow();
 
     if (message == QLatin1String("CHECKUPDATES"))
     {
@@ -399,8 +399,8 @@
     actWin->raise();
     if (actWin->isHidden())
       actWin->show();
-    else
-      actWin->activateWindow();
+    //else
+      //actWin->activateWindow();
 
     const QStringList packagesToInstallList =
         message.split(QLatin1Char(','), Qt::SkipEmptyParts);
@@ -415,8 +415,8 @@
         actWin->raise();
         if (actWin->isHidden())
           actWin->show();
-        else
-          actWin->activateWindow();
+        //else
+          //actWin->activateWindow();
     }
   else if (actWin && message == QLatin1String("HIDE")) {
     if (!actWin->isHidden())
