--- a/solid-device-automounter/lib/AutomounterSettingsBase.kcfg	2022-02-04 21:28:36.371348714 +0100
+++ b/solid-device-automounter/lib/AutomounterSettingsBase.kcfg	2022-02-04 21:28:36.435349358 +0100
@@ -15,7 +15,7 @@
         <default>false</default>
     </entry>
     <entry name="AutomountEnabled" type="Bool">
-        <default>true</default>
+        <default>false</default>
     </entry>
   </group>
 </kcfg>
