--- a/src/gui/widgets/Panner.cpp	2020-06-06 15:07:16.327232487 +0200
+++ b/src/gui/widgets/Panner.cpp	2020-06-06 15:07:16.413230548 +0200
@@ -26,6 +26,7 @@
 
 #include <QPolygon>
 #include <QMouseEvent>
+#include <QPainterPath>
 
 #include <iostream>
 

--- a/src/gui/general/ThornStyle.cpp	2020-06-06 15:28:35.836523735 +0200
+++ b/src/gui/general/ThornStyle.cpp	2020-06-06 15:28:35.930521634 +0200
@@ -34,6 +34,7 @@
 #include <QLabel>
 #include <QLayout>
 #include <QPainter>
+#include <QPainterPath>
 #include <QRadioButton>
 #include <QStyleFactory>
 #include <QStyleOption>
