--- marble-4.11.1/data/maps/earth/srtm/srtm.dgml.orig	2013-08-28 13:03:55.010869166 -0400
+++ marble-4.11.1/data/maps/earth/srtm/srtm.dgml	2013-09-14 18:51:30.864209791 -0400
@@ -15,7 +15,7 @@
             <description><![CDATA[<p>A <i>classic topographic map</i>.<p>It uses vector lines to mark coastlines, country borders etc. and bitmap graphics to create the height relief.]]></description>
     
             <zoom>
-                <minimum>   900  </minimum>
+                <minimum>  1060  </minimum>
                 <maximum>  2100  </maximum>
     
                 <!-- describes whether you can zoom trough the levels
