--- src/backends/x11/xlibbackend.cpp_orig	2014-02-11 21:00:25.252360806 -0500
+++ src/backends/x11/xlibbackend.cpp	2014-02-11 22:13:31.489859502 -0500
@@ -35,8 +35,8 @@
 #include <X11/extensions/XInput.h>
 #include <X11/extensions/XInput2.h>
 
-#include <synaptics-properties.h>
-#include <xserver-properties.h>
+#include </usr/include/xorg/synaptics-properties.h>
+#include </usr/include/xorg/xserver-properties.h>
 
 #include "synclientproperties.h"
 
