diff --git a/drkonqi/data/debuggers/internal/gdbrc b/drkonqi/data/debuggers/internal/gdbrc
index d998c66..c5c5b62 100644
--- a/drkonqi/data/debuggers/internal/gdbrc
+++ b/drkonqi/data/debuggers/internal/gdbrc
@@ -73,5 +73,5 @@ TryExec=gdb
 Backends=KCrash
 
 [KCrash]
-Exec=gdb -nw -n -batch -x %tempfile -p %pid %execpath
+Exec=gdb -nw -n -batch -ex "mt set target-async off" -ex "attach %pid" -x %tempfile %execpath
 BatchCommands=set width 200\nthread\nthread apply all bt
