commit 3b23c0aafe5bce8b443ce81d23601efe66e0e865
Author: Andreas Hartmetz <ahartmetz@gmail.com>
Date:   Wed May 14 05:16:04 2014 +0200

    Adapt to changes in KAboutData - one more file.

diff --git a/drkonqi/tests/bugzillalibtest/bugzillalibtest.cpp b/drkonqi/tests/bugzillalibtest/bugzillalibtest.cpp
index e0412e0..a10b876 100644
--- a/drkonqi/tests/bugzillalibtest/bugzillalibtest.cpp
+++ b/drkonqi/tests/bugzillalibtest/bugzillalibtest.cpp
@@ -127,8 +127,8 @@ class BugzillaLibTest : public QObject
 int main (int argc, char ** argv)
 {
     QApplication app(argc, argv);
-    KAboutData aboutData( "bzlibtest", "bzlibtest", i18n("BugzillaLib Test (DrKonqi2)"),
-        "1.0", i18n("Test application for bugtracker manager lib"), KAboutData::License_GPL,
+    KAboutData aboutData( "bzlibtest", i18n("BugzillaLib Test (DrKonqi2)"),
+        "1.0", i18n("Test application for bugtracker manager lib"), KAboutLicense::GPL,
         i18n("(c) 2009, DrKonqi2 Developers"));
 
     QCommandLineParser parser;
