diff --git a/src/streamrestore.cpp b/src/streamrestore.cpp
index f04b75a..4df5fa7 100644
--- a/src/streamrestore.cpp
+++ b/src/streamrestore.cpp
@@ -115,9 +115,13 @@ bool StreamRestore::isMuted() const
 void StreamRestore::setMuted(bool muted)
 {
     if (m_cache.valid) {
-        writeChanges(m_cache.volume, m_cache.channelMap, muted, m_cache.device);
+        if (m_cache.muted != muted) {
+            writeChanges(m_cache.volume, m_cache.channelMap, muted, m_cache.device);
+        }
     } else {
-        writeChanges(m_volume, m_channelMap, muted, m_device);
+        if (m_muted != muted) {
+            writeChanges(m_volume, m_channelMap, muted, m_device);
+        }
     }
 }
