--- a/src/lib/plugins/speeddial.cpp	2023-08-11 22:47:41.289155385 +0200
+++ b/src/lib/plugins/speeddial.cpp	2023-08-11 22:47:41.353156137 +0200
@@ -63,10 +63,12 @@
     settings.endGroup();
 
     if (allPages.isEmpty()) {
-        allPages = "url:\"https://www.falkon.org\"|title:\"Falkon\";"
-                   "url:\"https://store.falkon.org\"|title:\"Falkon Store\";"
-                   "url:\"https://www.kde.org\"|title:\"KDE Planet\";"
-                   "url:\"https://planet.kde.org\"|title:\"KDE Community\";";
+        allPages = "url:\"https://kaosx.us\"|title:\"KaOS - Welcome\";"
+                   "url:\"https://forum.kaosx.us\"|title:\"Forum\";"
+                   "url:\"https://webchat.kde.org/#/room/#KaOSx_KaOS:gitter.im\"|title:\"Matrix\";"
+                   "url:\"https://github.com/KaOSx\"|title:\"KaOS Github\";"
+                   "url:\"https://www.falkon.org\"|title:\"Falkon\";"
+                   "url:\"https://www.kde.org\"|title:\"KDE Community\";";
     }
     changed(allPages);
 
