--- a/src/lib/plugins/speeddial.cpp	2023-07-20 11:49:30.094965418 +0200
+++ b/src/lib/plugins/speeddial.cpp	2023-07-20 11:49:30.156966154 +0200
@@ -63,10 +63,12 @@
     settings.endGroup();
 
     if (allPages.isEmpty()) {
-        allPages = "url:\"https://www.falkon.org\"|title:\"Falkon\";"
+        allPages = "url:\"https://kaosx.us\"|title:\"KaOS - Welcome\";"
+                  "url:\"https://forum.kaosx.us\"|title:\"Forum\";"
+                  "url:\"https://web.libera.chat/#KaOS\"|title:\"IRC\";"
+                   "url:\"https://www.falkon.org\"|title:\"Falkon\";"
                    "url:\"https://store.falkon.org\"|title:\"Falkon Store\";"
-                   "url:\"https://www.kde.org\"|title:\"KDE Planet\";"
-                   "url:\"https://planet.kde.org\"|title:\"KDE Community\";";
+                   "url:\"https://www.kde.org\"|title:\"KDE Community\";";
     }
     changed(allPages);
 
