--- a/data/profiles/Shell.profile	2012-08-13 05:24:27.000000000 -0400
+++ b/data/profiles/Shell.profile	2012-09-03 19:15:44.689246940 -0400
@@ -170,3 +170,17 @@
 Comment[x-test]=xxKonsole default profilexx
 Comment[zh_CN]=Konsole 默认配置
 Comment[zh_TW]=Konsole 預設設定檔
+
+ShowNewAndCloseTabButtons=true
+
+[Appearance]
+AntiAliasFonts=true
+ColorScheme=Linux
+Font=Bitstream Vera Sans Mono,10,-1,0,75,0,0,0,0,0
+ 
+[Scrolling]
+HistoryMode=2
+
+[Terminal Features]
+BlinkingCursorEnabled=true
+BidiRenderingEnabled=true
